{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from Bio import SeqIO\n",
    "import re\n",
    "import math\n",
    "pd.set_option(\"display.max_rows\",101)\n",
    "pd.set_option(\"display.max_columns\",101)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Parsing the variants.csv file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/timothywannier/anaconda/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2717: DtypeWarning: Columns (23,35,71) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n",
      "/Users/timothywannier/anaconda/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2717: DtypeWarning: Columns (15,35,71) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "mgdata = pd.read_csv(\"variants_ALL4_to_MG1655v3.csv\",dtype = {'IS_HET': str, 'QA': str} )\n",
    "mgdata = mgdata.loc[:,['UID','POSITION','EXPERIMENT_SAMPLE_LABEL','REF','ALT','DP','AF','INFO_EFF_GENE']]\n",
    "mgdata = mgdata.rename(columns={'EXPERIMENT_SAMPLE_LABEL':'lineage','POSITION':'position','INFO_EFF_GENE':'gene'})\n",
    "mgdata = mgdata[mgdata.lineage.notnull()]\n",
    "c3data = pd.read_csv(\"variants_ALL4_to_C321.csv\",dtype = {'IS_HET': str, 'QA': str} )\n",
    "c3data = c3data.loc[:,['UID','POSITION','EXPERIMENT_SAMPLE_LABEL','REF','ALT','DP','AF','INFO_EFF_GENE']]\n",
    "c3data = c3data.rename(columns={'EXPERIMENT_SAMPLE_LABEL':'lineage','POSITION':'position','INFO_EFF_GENE':'gene'})\n",
    "c3data = c3data[c3data.lineage.notnull()]\n",
    "\n",
    "# reassign P populations because of mis-labeling of samples somehow\n",
    "pDic = {'P-1-1':'P-1-1','P-12-2':'P-1-2',\n",
    "        'P-1-2':'P-2-1','P-13-1':'P-2-2',\n",
    "        'P-2-1':'P-3-1','P-11-2':'P-3-2',\n",
    "        'P-2-2':'P-4-1','P-12-1':'P-4-2',\n",
    "        'P-3-1':'P-5-1','P-6-2':'P-5-2',\n",
    "        'P-3-2':'P-6-1','P-7-1':'P-6-2',\n",
    "        'P-4-1':'P-7-1','P-7-2':'P-7-2',\n",
    "        'P-4-2':'P-8-1','P-8-1':'P-8-2',\n",
    "        'P-5-1':'P-9-1','P-8-2':'P-9-2',\n",
    "        'P-5-2':'P-10-1','P-9-1':'P-10-2',\n",
    "        'P-6-1':'P-11-1','P-9-2':'P-11-2',\n",
    "        'P-10-1':'P-12-1','P-13-2':'P-12-2',\n",
    "        'P-10-2':'P-13-1','P-14-1':'P-13-2',\n",
    "        'P-11-1':'P-14-1','P-14-2':'P-14-2'}\n",
    "\n",
    "c3data['lineage'] = c3data['lineage'].apply(lambda x: pDic[x] if x in pDic.keys() else x)\n",
    "mgdata['lineage'] = mgdata['lineage'].apply(lambda x: pDic[x] if x in pDic.keys() else x)\n",
    "\n",
    "c_count = mgdata[mgdata.lineage.str[0] == 'C'].lineage.nunique()\n",
    "e_count = mgdata[mgdata.lineage.str[0] == 'E'].lineage.nunique()\n",
    "f_count = mgdata[mgdata.lineage.str[0] == 'F'].lineage.nunique()\n",
    "p_count = mgdata[mgdata.lineage.str[0] == 'P'].lineage.nunique()\n",
    "cfp_count = c_count + f_count + p_count\n",
    "c_samples = list(mgdata[mgdata.lineage.str[0] == 'C'].lineage.unique())\n",
    "e_samples = list(mgdata[mgdata.lineage.str[0] == 'E'].lineage.unique())\n",
    "f_samples = list(mgdata[mgdata.lineage.str[0] == 'F'].lineage.unique())\n",
    "p_samples = list(mgdata[mgdata.lineage.str[0] == 'P'].lineage.unique())\n",
    "cfp_samples = c_samples + f_samples + p_samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UID</th>\n",
       "      <th>position</th>\n",
       "      <th>lineage</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>DP</th>\n",
       "      <th>AF</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>C-C3-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>103.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>E-C1-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>C-C4-1</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>59.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>C-A4-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>50.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>C-E3-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>F-E6-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>69.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>P-2-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>57.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>P-13-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>C-F3-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>F-G5-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>69.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>F-G5-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>82.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>E-G0</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>E-A2-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>F-B6-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>F-C6-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>97.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>P-12-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>P-5-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>P-4-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>C-G4-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>94.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>E-D1-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>E-E2-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>87.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>F-D6-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>73.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>C-D4-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>67.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>F-H5-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>74.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>P-14-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>E-D2-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>79.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>E-D2-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>87.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>C-D4-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>95.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>P-5-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>P-14-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>E-A2-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>90.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>F-H5-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>81.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>C-E4-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>76.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>C-G3-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>56.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>E-E1-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>84.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>C-F3-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>114.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>E-F1-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>E-G1-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>C-E4-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>92.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>E-F2-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>84.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>P-6-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>P-7-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>E-E2-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>77.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>E-H1-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>61.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>W72</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>80.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>C-F4-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>77.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>F-D5-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>89.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>P-8-2</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>E-H2-1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>77.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>28ee39d1</td>\n",
       "      <td>84</td>\n",
       "      <td>C-B4-1</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>62.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522667</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>F-G6-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>65.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522668</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>P-10-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522669</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-C3-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>62.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522670</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>P-9-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522671</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>P-8-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>73.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522672</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>P-3-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>59.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522673</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>P-7-2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>arcA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522674</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>P-12-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522675</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>E-E1-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522676</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-G3-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522677</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-F3-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>72.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522678</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-E3-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>42.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522679</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-C4-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522680</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-G3-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522681</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-G0</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522682</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-G4-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>53.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522683</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-A4-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522684</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-B3-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522685</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-C4-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>59.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522686</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>E-E1-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>51.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522687</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>E-C1-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522688</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-A4-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522689</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-B4-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522690</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>P-8-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522691</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>E-D2-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>45.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522692</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>W62</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522693</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>E-D1-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522694</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>E-G0</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522695</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>P-10-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>102.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522696</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>E-E2-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522697</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>E-A2-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522698</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>E-C2-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522699</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>E-D2-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>86.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522700</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-E4-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>55.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522701</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-G4-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>51.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522702</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>W72</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522703</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>E-A1-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522704</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-D3-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522705</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>F-E6-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>63.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522706</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>E-A1-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522707</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-D3-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522708</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-F4-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522709</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-F3-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522710</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>E-H2-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522711</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>E-B2-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522712</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>F-E6-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522713</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-E3-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>58.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522714</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>E-D1-1</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522715</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>C-F4-2</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522716</th>\n",
       "      <td>e0e6e095</td>\n",
       "      <td>4639581</td>\n",
       "      <td>P-7-1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>arcA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>519940 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             UID  position lineage REF  ALT     DP   AF  gene\n",
       "0       28ee39d1        84  C-C3-1   A  NaN  103.0  0.0   NaN\n",
       "2       28ee39d1        84  E-C1-1   A  NaN   30.0  0.0   NaN\n",
       "3       28ee39d1        84  C-C4-1   A    G   59.0  1.0   NaN\n",
       "4       28ee39d1        84  C-A4-2   A  NaN   50.0  0.0   NaN\n",
       "5       28ee39d1        84  C-E3-1   A  NaN   75.0  0.0   NaN\n",
       "6       28ee39d1        84  F-E6-2   A  NaN   69.0  0.0   NaN\n",
       "7       28ee39d1        84   P-2-1   A  NaN   57.0  0.0   NaN\n",
       "8       28ee39d1        84  P-13-1   A  NaN   27.0  0.0   NaN\n",
       "9       28ee39d1        84  C-F3-1   A  NaN   41.0  0.0   NaN\n",
       "10      28ee39d1        84  F-G5-1   A  NaN   69.0  0.0   NaN\n",
       "11      28ee39d1        84  F-G5-2   A  NaN   82.0  0.0   NaN\n",
       "12      28ee39d1        84    E-G0   A  NaN   32.0  0.0   NaN\n",
       "13      28ee39d1        84  E-A2-1   A  NaN   54.0  0.0   NaN\n",
       "14      28ee39d1        84  F-B6-1   A  NaN   48.0  0.0   NaN\n",
       "15      28ee39d1        84  F-C6-2   A  NaN   97.0  0.0   NaN\n",
       "16      28ee39d1        84  P-12-2   A  NaN   25.0  0.0   NaN\n",
       "17      28ee39d1        84   P-5-1   A  NaN   20.0  0.0   NaN\n",
       "18      28ee39d1        84   P-4-1   A  NaN   38.0  0.0   NaN\n",
       "19      28ee39d1        84  C-G4-1   A  NaN   94.0  0.0   NaN\n",
       "20      28ee39d1        84  E-D1-2   A  NaN   27.0  0.0   NaN\n",
       "21      28ee39d1        84  E-E2-1   A  NaN   87.0  0.0   NaN\n",
       "22      28ee39d1        84  F-D6-1   A  NaN   73.0  0.0   NaN\n",
       "23      28ee39d1        84  C-D4-1   A  NaN   67.0  0.0   NaN\n",
       "24      28ee39d1        84  F-H5-2   A  NaN   74.0  0.0   NaN\n",
       "25      28ee39d1        84  P-14-2   A  NaN   29.0  0.0   NaN\n",
       "26      28ee39d1        84  E-D2-1   A  NaN   79.0  0.0   NaN\n",
       "27      28ee39d1        84  E-D2-2   A  NaN   87.0  0.0   NaN\n",
       "28      28ee39d1        84  C-D4-2   A  NaN   95.0  0.0   NaN\n",
       "29      28ee39d1        84   P-5-2   A  NaN   15.0  0.0   NaN\n",
       "30      28ee39d1        84  P-14-1   A  NaN   36.0  0.0   NaN\n",
       "31      28ee39d1        84  E-A2-2   A  NaN   90.0  0.0   NaN\n",
       "32      28ee39d1        84  F-H5-1   A  NaN   81.0  0.0   NaN\n",
       "33      28ee39d1        84  C-E4-1   A  NaN   76.0  0.0   NaN\n",
       "34      28ee39d1        84  C-G3-1   A  NaN   56.0  0.0   NaN\n",
       "35      28ee39d1        84  E-E1-2   A  NaN   84.0  0.0   NaN\n",
       "36      28ee39d1        84  C-F3-2   A  NaN  114.0  0.0   NaN\n",
       "37      28ee39d1        84  E-F1-1   A  NaN   44.0  0.0   NaN\n",
       "38      28ee39d1        84  E-G1-1   A  NaN   35.0  0.0   NaN\n",
       "39      28ee39d1        84  C-E4-2   A  NaN   92.0  0.0   NaN\n",
       "40      28ee39d1        84  E-F2-2   A  NaN   84.0  0.0   NaN\n",
       "41      28ee39d1        84   P-6-2   A  NaN   34.0  0.0   NaN\n",
       "42      28ee39d1        84   P-7-1   A  NaN   48.0  0.0   NaN\n",
       "43      28ee39d1        84  E-E2-2   A  NaN   77.0  0.0   NaN\n",
       "44      28ee39d1        84  E-H1-2   A  NaN   61.0  0.0   NaN\n",
       "45      28ee39d1        84     W72   A  NaN   80.0  0.0   NaN\n",
       "46      28ee39d1        84  C-F4-2   A  NaN   77.0  0.0   NaN\n",
       "47      28ee39d1        84  F-D5-1   A  NaN   89.0  0.0   NaN\n",
       "48      28ee39d1        84   P-8-2   A  NaN   40.0  0.0   NaN\n",
       "49      28ee39d1        84  E-H2-1   A  NaN   77.0  0.0   NaN\n",
       "50      28ee39d1        84  C-B4-1   A    G   62.0  1.0   NaN\n",
       "...          ...       ...     ...  ..  ...    ...  ...   ...\n",
       "522667  e0e6e095   4639581  F-G6-1   T  NaN   65.0  0.0   NaN\n",
       "522668  e0e6e095   4639581  P-10-2   T  NaN   28.0  0.0   NaN\n",
       "522669  e0e6e095   4639581  C-C3-2   T  NaN   62.0  0.0   NaN\n",
       "522670  e0e6e095   4639581   P-9-2   T  NaN   16.0  0.0   NaN\n",
       "522671  e0e6e095   4639581   P-8-1   T  NaN   73.0  0.0   NaN\n",
       "522672  e0e6e095   4639581   P-3-1   T  NaN   59.0  0.0   NaN\n",
       "522673  e0e6e095   4639581   P-7-2   T    C   14.0  1.0  arcA\n",
       "522674  e0e6e095   4639581  P-12-2   T  NaN   27.0  0.0   NaN\n",
       "522675  e0e6e095   4639581  E-E1-1   T  NaN   18.0  0.0   NaN\n",
       "522676  e0e6e095   4639581  C-G3-1   T  NaN   28.0  0.0   NaN\n",
       "522677  e0e6e095   4639581  C-F3-2   T  NaN   72.0  0.0   NaN\n",
       "522678  e0e6e095   4639581  C-E3-1   T  NaN   42.0  0.0   NaN\n",
       "522679  e0e6e095   4639581  C-C4-1   T  NaN   26.0  0.0   NaN\n",
       "522680  e0e6e095   4639581  C-G3-2   T  NaN   30.0  0.0   NaN\n",
       "522681  e0e6e095   4639581    C-G0   T  NaN   35.0  0.0   NaN\n",
       "522682  e0e6e095   4639581  C-G4-1   T  NaN   53.0  0.0   NaN\n",
       "522683  e0e6e095   4639581  C-A4-2   T  NaN   25.0  0.0   NaN\n",
       "522684  e0e6e095   4639581  C-B3-1   T  NaN  100.0  0.0   NaN\n",
       "522685  e0e6e095   4639581  C-C4-2   T  NaN   59.0  0.0   NaN\n",
       "522686  e0e6e095   4639581  E-E1-2   T  NaN   51.0  0.0   NaN\n",
       "522687  e0e6e095   4639581  E-C1-2   T  NaN   17.0  0.0   NaN\n",
       "522688  e0e6e095   4639581  C-A4-1   T  NaN   26.0  0.0   NaN\n",
       "522689  e0e6e095   4639581  C-B4-1   T  NaN   21.0  0.0   NaN\n",
       "522690  e0e6e095   4639581   P-8-2   T  NaN   31.0  0.0   NaN\n",
       "522691  e0e6e095   4639581  E-D2-2   T  NaN   45.0  0.0   NaN\n",
       "522692  e0e6e095   4639581     W62   T  NaN   30.0  0.0   NaN\n",
       "522693  e0e6e095   4639581  E-D1-2   T  NaN   10.0  0.0   NaN\n",
       "522694  e0e6e095   4639581    E-G0   T  NaN   26.0  0.0   NaN\n",
       "522695  e0e6e095   4639581  P-10-1   T  NaN  102.0  0.0   NaN\n",
       "522696  e0e6e095   4639581  E-E2-2   T  NaN   39.0  0.0   NaN\n",
       "522697  e0e6e095   4639581  E-A2-1   T  NaN    7.0  0.0   NaN\n",
       "522698  e0e6e095   4639581  E-C2-2   T  NaN   29.0  0.0   NaN\n",
       "522699  e0e6e095   4639581  E-D2-1   T  NaN   86.0  0.0   NaN\n",
       "522700  e0e6e095   4639581  C-E4-2   T  NaN   55.0  0.0   NaN\n",
       "522701  e0e6e095   4639581  C-G4-2   T  NaN   51.0  0.0   NaN\n",
       "522702  e0e6e095   4639581     W72   T  NaN   29.0  0.0   NaN\n",
       "522703  e0e6e095   4639581  E-A1-2   T  NaN   11.0  0.0   NaN\n",
       "522704  e0e6e095   4639581  C-D3-1   T  NaN   30.0  0.0   NaN\n",
       "522705  e0e6e095   4639581  F-E6-1   T  NaN   63.0  0.0   NaN\n",
       "522706  e0e6e095   4639581  E-A1-1   T  NaN    2.0  0.0   NaN\n",
       "522707  e0e6e095   4639581  C-D3-2   T  NaN   37.0  0.0   NaN\n",
       "522708  e0e6e095   4639581  C-F4-1   T  NaN    5.0  0.0   NaN\n",
       "522709  e0e6e095   4639581  C-F3-1   T  NaN   21.0  0.0   NaN\n",
       "522710  e0e6e095   4639581  E-H2-1   T  NaN   38.0  0.0   NaN\n",
       "522711  e0e6e095   4639581  E-B2-1   T  NaN   39.0  0.0   NaN\n",
       "522712  e0e6e095   4639581  F-E6-2   T  NaN   33.0  0.0   NaN\n",
       "522713  e0e6e095   4639581  C-E3-2   T  NaN   58.0  0.0   NaN\n",
       "522714  e0e6e095   4639581  E-D1-1   T  NaN   20.0  0.0   NaN\n",
       "522715  e0e6e095   4639581  C-F4-2   T  NaN   34.0  0.0   NaN\n",
       "522716  e0e6e095   4639581   P-7-1   T    C   70.0  1.0  arcA\n",
       "\n",
       "[519940 rows x 8 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgdata"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure out a bit of stuff about the variant set....looking for numbers of certain types of variants, read depth, etc.\n",
    "\n",
    "Instance 1: Number of variant calls with read depth < min_read_depth\n",
    "Instance 2: Number of variant calls with allele fracgion < min_allele_fraction "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total variant calls to MG1655v3:  68415 \n",
      " -- at an average read depth of:  52.8\n",
      "variant calls under min read depth:  2364\n",
      "variant calls under min allele fraction:  1345\n",
      "\n",
      "total variant calls to c321 genbank:  33225 \n",
      " -- at an average read depth of:  53.2\n",
      "variant calls under min read depth:  812\n",
      "variant calls under min allele fraction:  1089\n",
      "total MG1655 culled variant calls:  64639\n",
      "total C321 culled variant calls:  31254\n",
      "\n",
      "total gen0 MG1655 ENCR2.1 culled variant calls:  28\n",
      "total gen0 MG1655 C321.∆A culled variant calls:  685\n",
      "total gen0 MG1655 C321.∆A_F11 culled variant calls:  714\n",
      "\n",
      "total MG1655 to ENCR2.1 culled variant calls: 1919  -- per sample:  68.5  -- and in excess of gen0:  40.5 \n",
      "   total unique variants:  818  -- not found in gen0:  796\n",
      "total MG1655 to C321 culled variant calls: 20671  -- per sample:  738.2\n",
      "total MG1655 to C321.∆A culled variant calls: 19251  -- per sample:  740.4  -- and in excess of gen0:  55.4 \n",
      "   total unique variants:  1580  -- not found in gen0:  897\n",
      "total MG1655 to C321.∆A_F11 culled variant calls: 21329  -- per sample:  761.8  -- and in excess of gen0:  47.8 \n",
      "   total unique variants:  1548  -- not found in gen0:  836\n",
      "\n",
      "total gen0 C321 ENCR2.1 culled variant calls:  681\n",
      "total gen0 C321 C321.∆A culled variant calls:  26\n",
      "total gen0 C321 C321.∆A_F11 culled variant calls:  86\n",
      "\n",
      "total C321 to ENCR2.1 culled variant calls: 20218  -- per sample:  722.1  -- and in excess of gen0:  41.1 \n",
      "   total unique variants:  1475  -- not found in gen0:  802\n",
      "total C321 to C321 culled variant calls: 2951  -- per sample:  105.4\n",
      "total C321 to C321.∆A culled variant calls: 2208  -- per sample:  84.9  -- and in excess of gen0:  58.9 \n",
      "   total unique variants:  918  -- not found in gen0:  894\n",
      "total C321 to C321.∆A_F11 culled variant calls: 3713  -- per sample:  132.6  -- and in excess of gen0:  46.6 \n",
      "   total unique variants:  922  -- not found in gen0:  836\n"
     ]
    }
   ],
   "source": [
    "# Set parameters for judging variant call quality. 1. Length of reference, 2. read depth, 3. allele fraction\n",
    "# Apply more stringent criteria to variants with long references, often which have strings of single nucleotides.\n",
    "# Also weed out variants with high SRP (uneven coverage on F and R strands)?\n",
    "long = 6\n",
    "depth = 6\n",
    "depth_long = 6\n",
    "fraction = 0.9\n",
    "fraction_long = 0.9\n",
    "#max_srp = 100\n",
    "\n",
    "# create dataframes that contain only the called variants\n",
    "mv = mgdata[mgdata.ALT.notnull()]\n",
    "cv = c3data[c3data.ALT.notnull()]\n",
    "\n",
    "# report out some statistics\n",
    "print(\"total variant calls to MG1655v3: \",mv.UID.count(),\n",
    "      \"\\n -- at an average read depth of: \", round(mgdata.DP.mean(),1))\n",
    "print(\"variant calls under min read depth: \",mv[mv.DP < depth].UID.count())\n",
    "print(\"variant calls under min allele fraction: \",mv[mv.AF < fraction].UID.count())\n",
    "print(\"\\ntotal variant calls to c321 genbank: \",cv.UID.count(),\n",
    "      \"\\n -- at an average read depth of: \", round(c3data.DP.mean(),1))\n",
    "print(\"variant calls under min read depth: \",cv[cv.DP < depth].UID.count())\n",
    "print(\"variant calls under min allele fraction: \",cv[cv.AF < fraction].UID.count())\n",
    "\n",
    "# Remove reads under minimal read depth or allele fraction. \n",
    "mvc = mv[(((mv.DP >= depth) & (mv.REF.str.len() < long)) | ((mv.DP >= depth_long) & (mv.REF.str.len() > long))) &\n",
    "         (((mv.AF >= fraction) & (mv.REF.str.len() < long)) | ((mv.AF >= fraction_long) & (mv.REF.str.len() > long)))]       \n",
    "                 \n",
    "cvc = cv[(((cv.DP >= depth) & (cv.REF.str.len() < long)) | ((cv.DP >= depth_long) & (cv.REF.str.len() > long))) &\n",
    "         (((cv.AF >= fraction) & (cv.REF.str.len() < long)) | ((cv.AF >= fraction_long) & (cv.REF.str.len() > long)))]       \n",
    "\n",
    "print(\"total MG1655 culled variant calls: \",mvc.UID.count())\n",
    "print(\"total C321 culled variant calls: \",cvc.UID.count())\n",
    "\n",
    "# Separate the culled variant calls by..\n",
    "# data set: ECNR2.1, C321, C321.∆A, and C321.∆A_F11\n",
    "# generation: g0 vs g1,100\n",
    "# alignment: to MG1655 or to C321.deltaA\n",
    "\n",
    "g0Emvc = mvc[(mvc.lineage.str[3] == '0') & (mvc.lineage.str[0] == 'E')]\n",
    "g0Cmvc = mvc[(mvc.lineage.str[3] == '0') & (mvc.lineage.str[0] == 'C')]\n",
    "g0Fmvc = mvc[(mvc.lineage.str[3] == '0') & (mvc.lineage.str[0] == 'F')]\n",
    "Emvc = mvc[(mvc.lineage.str[3] != '0') & (mvc.lineage.str[0] == 'E')]\n",
    "Cmvc = mvc[(mvc.lineage.str[3] != '0') & (mvc.lineage.str[0] == 'C')]\n",
    "Fmvc = mvc[(mvc.lineage.str[3] != '0') & (mvc.lineage.str[0] == 'F')]\n",
    "Pmvc = mvc[mvc.lineage.str[0] == 'P']\n",
    "g0Ecvc = cvc[(cvc.lineage.str[3] == '0') & (cvc.lineage.str[0] == 'E')]\n",
    "g0Ccvc = cvc[(cvc.lineage.str[3] == '0') & (cvc.lineage.str[0] == 'C')]\n",
    "g0Fcvc = cvc[(cvc.lineage.str[3] == '0') & (cvc.lineage.str[0] == 'F')]\n",
    "Ecvc = cvc[(cvc.lineage.str[3] != '0') & (cvc.lineage.str[0] == 'E')]\n",
    "Ccvc = cvc[(cvc.lineage.str[3] != '0') & (cvc.lineage.str[0] == 'C')]\n",
    "Fcvc = cvc[(cvc.lineage.str[3] != '0') & (cvc.lineage.str[0] == 'F')]\n",
    "Pcvc = cvc[cvc.lineage.str[0] == 'P']\n",
    "\n",
    "# print out some statistics\n",
    "print(\"\\ntotal gen0 MG1655 ENCR2.1 culled variant calls: \",g0Emvc.UID.count())\n",
    "print(\"total gen0 MG1655 C321.∆A culled variant calls: \",g0Cmvc.UID.count())\n",
    "print(\"total gen0 MG1655 C321.∆A_F11 culled variant calls: \",g0Fmvc.UID.count())\n",
    "print(\"\\ntotal MG1655 to ENCR2.1 culled variant calls:\",Emvc.UID.count(),\" -- per sample: \",\n",
    "      round(Emvc.UID.count() / Emvc.lineage.nunique(),1),\n",
    "      \" -- and in excess of gen0: \",\n",
    "      round(Emvc.UID.count() / Emvc.lineage.nunique()-g0Emvc.UID.count(),1),\n",
    "      \"\\n   total unique variants: \",Emvc.UID.nunique(),\" -- not found in gen0: \",\n",
    "      Emvc.UID[~Emvc.UID.isin(g0Emvc.UID)].nunique())\n",
    "print(\"total MG1655 to C321 culled variant calls:\",Pmvc.UID.count(),\" -- per sample: \",\n",
    "      round(Pmvc.UID.count() / Pmvc.lineage.nunique(),1))\n",
    "print(\"total MG1655 to C321.∆A culled variant calls:\",Cmvc.UID.count(),\" -- per sample: \",\n",
    "      round(Cmvc.UID.count() / Cmvc.lineage.nunique(),1),\n",
    "      \" -- and in excess of gen0: \",\n",
    "      round(Cmvc.UID.count() / Cmvc.lineage.nunique()-g0Cmvc.UID.count(),1),\n",
    "      \"\\n   total unique variants: \",Cmvc.UID.nunique(),\" -- not found in gen0: \",\n",
    "      Cmvc.UID[~Cmvc.UID.isin(g0Cmvc.UID)].nunique())\n",
    "print(\"total MG1655 to C321.∆A_F11 culled variant calls:\",Fmvc.UID.count(),\" -- per sample: \",\n",
    "      round(Fmvc.UID.count() / Fmvc.lineage.nunique(),1),\n",
    "      \" -- and in excess of gen0: \",\n",
    "      round(Fmvc.UID.count() / Fmvc.lineage.nunique()-g0Fmvc.UID.count(),1),\n",
    "      \"\\n   total unique variants: \",Fmvc.UID.nunique(),\" -- not found in gen0: \",\n",
    "      Fmvc.UID[~Fmvc.UID.isin(g0Fmvc.UID)].nunique())\n",
    "print(\"\\ntotal gen0 C321 ENCR2.1 culled variant calls: \",g0Ecvc.UID.count())\n",
    "#print(\"total gen0 C321 C321 culled variant calls: \",g0Pcvc.UID.count())\n",
    "print(\"total gen0 C321 C321.∆A culled variant calls: \",g0Ccvc.UID.count())\n",
    "print(\"total gen0 C321 C321.∆A_F11 culled variant calls: \",g0Fcvc.UID.count())\n",
    "print(\"\\ntotal C321 to ENCR2.1 culled variant calls:\",Ecvc.UID.count(),\" -- per sample: \",\n",
    "      round(Ecvc.UID.count() / Ecvc.lineage.nunique(),1),\n",
    "      \" -- and in excess of gen0: \",\n",
    "      round(Ecvc.UID.count() / Ecvc.lineage.nunique()-g0Ecvc.UID.count(),1),\n",
    "      \"\\n   total unique variants: \",Ecvc.UID.nunique(),\" -- not found in gen0: \",\n",
    "      Ecvc.UID[~Ecvc.UID.isin(g0Ecvc.UID)].nunique())\n",
    "print(\"total C321 to C321 culled variant calls:\",Pcvc.UID.count(),\" -- per sample: \",\n",
    "      round(Pcvc.UID.count() / Pcvc.lineage.nunique(),1))\n",
    "print(\"total C321 to C321.∆A culled variant calls:\",Ccvc.UID.count(),\" -- per sample: \",\n",
    "      round(Ccvc.UID.count() / Ccvc.lineage.nunique(),1),\n",
    "      \" -- and in excess of gen0: \",\n",
    "      round(Ccvc.UID.count() / Ccvc.lineage.nunique()-g0Ccvc.UID.count(),1),\n",
    "      \"\\n   total unique variants: \",Ccvc.UID.nunique(),\" -- not found in gen0: \",\n",
    "      Ccvc.UID[~Ccvc.UID.isin(g0Ccvc.UID)].nunique())\n",
    "print(\"total C321 to C321.∆A_F11 culled variant calls:\",Fcvc.UID.count(),\" -- per sample: \",\n",
    "      round(Fcvc.UID.count() / Fcvc.lineage.nunique(),1),\n",
    "      \" -- and in excess of gen0: \",\n",
    "      round(Fcvc.UID.count() / Fcvc.lineage.nunique()-g0Fcvc.UID.count(),1),\n",
    "      \"\\n   total unique variants: \",Fcvc.UID.nunique(),\" -- not found in gen0: \",\n",
    "      Fcvc.UID[~Fcvc.UID.isin(g0Fcvc.UID)].nunique())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove gen0 variants from the evolved populations.  Then apply a threshold cutoff to look at interesting variants.  Only look at genes his multiple times across biological replicates.\n",
    "\n",
    "**We haven't yet sequenced c321_prfA+ generation zero, so for now, use c321.∆A as a stand in for this variant. Add in hand-called list of ancestral mutations here..\n",
    "\n",
    "(yaiO is a string of 9 Gs, so it's just miscalled)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "removed 632 ancestral hits from evolved populations.\n",
      "\n",
      "number of genes hit over threshold in evolved \n",
      "ECNR2.1 populations: 20  --C321 poipulations: 46  --C321.∆A poipulations: 36  --C321.∆A_F11 populations: 22 \n",
      "\n",
      "genes hit: \n",
      "        ECNR2  C321  C321_dA  C321_dA_F11\n",
      "aceE     NaN   1.0      7.0          2.0\n",
      "aceF     1.0   5.0      NaN          NaN\n",
      "acnA     NaN   NaN      5.0          2.0\n",
      "aer      1.0   NaN      5.0          2.0\n",
      "astD     NaN   5.0      NaN          NaN\n",
      "cmtA     NaN   NaN      NaN         10.0\n",
      "cyaA     NaN   4.0      5.0          3.0\n",
      "dcp      2.0   NaN      NaN          8.0\n",
      "dcuS    23.0   NaN      NaN          1.0\n",
      "dsdC     NaN   NaN      5.0          NaN\n",
      "efeO     NaN   NaN      NaN         11.0\n",
      "fdnG     NaN   NaN      5.0          NaN\n",
      "fimH     1.0   2.0      3.0         19.0\n",
      "fis      NaN   7.0     19.0          NaN\n",
      "flgK     NaN  27.0      1.0          NaN\n",
      "flu     31.0   NaN     11.0         22.0\n",
      "folA     NaN  22.0     20.0          4.0\n",
      "folM     NaN  10.0      1.0          NaN\n",
      "fryA     NaN   NaN      NaN          8.0\n",
      "glrR     NaN   6.0      NaN          NaN\n",
      "gltB     NaN   5.0      6.0          NaN\n",
      "hsdR     2.0  29.0      NaN          2.0\n",
      "insB1    2.0  24.0      3.0          2.0\n",
      "kup      1.0   5.0      3.0          NaN\n",
      "leuP     7.0   1.0      9.0          4.0\n",
      "leuQ     2.0   NaN      3.0          5.0\n",
      "leuV     8.0   NaN      NaN          7.0\n",
      "malK     NaN   NaN      5.0          2.0\n",
      "mdtJ     NaN   NaN      2.0          6.0\n",
      "menE     NaN   6.0      NaN          NaN\n",
      "mglA     1.0   6.0      NaN          NaN\n",
      "mtr     10.0   3.0      1.0          NaN\n",
      "mukB     NaN   2.0      5.0          NaN\n",
      "mutS     NaN   5.0      NaN          NaN\n",
      "narX     5.0   NaN      2.0          NaN\n",
      "nfrA     NaN  31.0      NaN          NaN\n",
      "nusA     NaN   6.0      2.0          NaN\n",
      "ompT    10.0   2.0      NaN          1.0\n",
      "oxyR     NaN   NaN      7.0         22.0\n",
      "parE     NaN   6.0      2.0          NaN\n",
      "pdxA     NaN  27.0      NaN          NaN\n",
      "pflB     NaN   5.0      1.0          NaN\n",
      "pfo      3.0   1.0      6.0          NaN\n",
      "phoP     6.0   NaN      1.0          9.0\n",
      "prc      NaN   6.0      1.0          NaN\n",
      "prfB     2.0   NaN     14.0         22.0\n",
      "prfC     NaN   NaN      3.0          5.0\n",
      "pta      NaN  10.0      1.0          NaN\n",
      "ptrB     3.0   1.0      8.0          NaN\n",
      "purL     2.0   6.0      2.0          NaN\n",
      "purT     NaN   8.0      4.0         14.0\n",
      "purU    26.0   2.0      NaN          NaN\n",
      "pykF    14.0   7.0      1.0          3.0\n",
      "pyrC    10.0   2.0      NaN          NaN\n",
      "pyrE    30.0  15.0     19.0         14.0\n",
      "recE     6.0   2.0      NaN          3.0\n",
      "rhsD     1.0   1.0      5.0          NaN\n",
      "rlmB     3.0   5.0      2.0          6.0\n",
      "rnlA     NaN   NaN      6.0          NaN\n",
      "rnr      NaN   6.0      NaN          NaN\n",
      "rph      2.0   6.0      8.0         16.0\n",
      "rpoB    13.0   9.0      3.0          NaN\n",
      "rpoC    18.0  16.0      8.0          3.0\n",
      "rsxC     1.0   7.0      NaN          2.0\n",
      "rsxE     NaN  27.0      NaN          NaN\n",
      "rusA    26.0   NaN      NaN          NaN\n",
      "rzpR     1.0  25.0      NaN          NaN\n",
      "slt      NaN   NaN      5.0          1.0\n",
      "speA     1.0   1.0      5.0          2.0\n",
      "thiD     5.0   NaN      2.0          NaN\n",
      "tyrR     NaN   NaN      4.0         10.0\n",
      "waaH     NaN   NaN      NaN          8.0\n",
      "wzzE     1.0   6.0      NaN          NaN\n",
      "xdhB     NaN   1.0      5.0          NaN\n",
      "yahL     NaN   NaN      9.0          2.0\n",
      "yaiO     NaN   NaN      NaN         13.0\n",
      "ybhG     NaN   3.0      6.0          NaN\n",
      "yccS     4.0   7.0      3.0          NaN\n",
      "ycgB     NaN   1.0      8.0          NaN\n",
      "ycjM     NaN   6.0      NaN          2.0\n",
      "ydfI     5.0   2.0      NaN          NaN\n",
      "ydfJ     1.0   NaN      7.0          2.0\n",
      "yeaJ     NaN  29.0      NaN          NaN\n",
      "yebB     NaN  24.0      NaN          NaN\n",
      "yeeJ     NaN   NaN      9.0          4.0\n",
      "yehB     1.0   NaN      9.0          NaN\n",
      "yeiE    19.0   NaN      2.0          NaN\n",
      "yejO     4.0  30.0      5.0          NaN\n",
      "yfcU     NaN   5.0      2.0          NaN\n",
      "yfhM     3.0   6.0      1.0          1.0\n",
      "yfiH     6.0   8.0      6.0          NaN\n",
      "ygcW     NaN   6.0      NaN          NaN\n",
      "ygeR     NaN   NaN      1.0          5.0\n",
      "yhjK     NaN   1.0      9.0          NaN\n",
      "yihT     NaN   NaN      5.0          NaN\n",
      "yjiN     1.0  29.0      NaN          2.0\n",
      "yjjU     NaN   NaN      6.0          NaN\n",
      "yliI     NaN   6.0      NaN          NaN\n",
      "yneO     4.0   9.0      3.0          6.0\n"
     ]
    }
   ],
   "source": [
    "# set threshold for minimum number of mutations needed to meet significance.\n",
    "threshold = 5\n",
    "positional_threshold = 3\n",
    "\n",
    "# remove UIDs that are present in the full variant list of the generation zero population \n",
    "g0_E_ALL = mv[(mv.lineage.str[3] == '0') & (mv.lineage.str[0] == 'E')]\n",
    "g0_C_ALL = mv[(mv.lineage.str[3] == '0') & (mv.lineage.str[0] == 'C')]\n",
    "g0_F_ALL = mv[(mv.lineage.str[3] == '0') & (mv.lineage.str[0] == 'F')]\n",
    "\n",
    "EmvcNEW = Emvc[~Emvc.UID.isin(g0_E_ALL.UID)]\n",
    "PmvcNEW = Pmvc[~Pmvc.UID.isin(g0_C_ALL.UID)] # use C321.dA g0 until c321 g0 is sequenced\n",
    "CmvcNEW = Cmvc[~Cmvc.UID.isin(g0_C_ALL.UID)]\n",
    "FmvcNEW = Fmvc[~Fmvc.UID.isin(g0_F_ALL.UID)] \n",
    "\n",
    "# crate output matrix of ancestral variants for each population\n",
    "ancestral_variants = pd.concat([Emvc[Emvc.UID.isin(g0_E_ALL.UID)].gene.value_counts(),\n",
    "                                Pmvc[Pmvc.UID.isin(g0_C_ALL.UID)].gene.value_counts(),\n",
    "                                Cmvc[Cmvc.UID.isin(g0_C_ALL.UID)].gene.value_counts(),\n",
    "                                Fmvc[Fmvc.UID.isin(g0_F_ALL.UID)].gene.value_counts()], axis=1)\n",
    "ancestral_variants.columns = ['ECNR2','C321','C321_dA','C321_dA_F11']\n",
    "\n",
    "# create list of new variants and filter for variants that occur at or above threshold\n",
    "new_hits = pd.concat([EmvcNEW.gene.value_counts(),PmvcNEW.gene.value_counts(),\n",
    "                      CmvcNEW.gene.value_counts(),FmvcNEW.gene.value_counts()], axis=1)\n",
    "new_hits.columns = ['ECNR2','C321','C321_dA','C321_dA_F11']\n",
    "threshold_hits = new_hits[(new_hits.ECNR2 >= threshold) | (new_hits.C321 >= threshold) | \n",
    "                          (new_hits.C321_dA >= threshold) | (new_hits.C321_dA_F11 >= threshold)]\n",
    "\n",
    "# print out some statistics\n",
    "print('removed',len(ancestral_variants), 'ancestral hits from evolved populations.\\n',)\n",
    "print('number of genes hit over threshold in evolved \\nECNR2.1 populations:',\n",
    "      threshold_hits[threshold_hits.ECNR2 >= threshold].ECNR2.count(),\n",
    "      ' --C321 poipulations:',\n",
    "      threshold_hits[threshold_hits.C321 >= threshold].C321.count(),\n",
    "      ' --C321.∆A poipulations:',\n",
    "      threshold_hits[threshold_hits.C321_dA >= threshold].C321_dA.count(),\n",
    "      ' --C321.∆A_F11 populations:',\n",
    "      threshold_hits[threshold_hits.C321_dA_F11 >= threshold].C321_dA_F11.count(),'\\n')\n",
    "print('genes hit: \\n',threshold_hits)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further analyze threshold variants, specifically looking for mutations in long strings of single nucleotides that could be sequencing errors.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# Building up a phylogenetic tree for each experiment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>C-A4-1</th>\n",
       "      <th>C-A4-2</th>\n",
       "      <th>C-B3-1</th>\n",
       "      <th>C-B3-2</th>\n",
       "      <th>C-B4-1</th>\n",
       "      <th>C-B4-2</th>\n",
       "      <th>C-C3-1</th>\n",
       "      <th>C-C3-2</th>\n",
       "      <th>C-C4-1</th>\n",
       "      <th>C-C4-2</th>\n",
       "      <th>C-D3-1</th>\n",
       "      <th>C-D3-2</th>\n",
       "      <th>C-D4-1</th>\n",
       "      <th>C-D4-2</th>\n",
       "      <th>C-E3-1</th>\n",
       "      <th>C-E3-2</th>\n",
       "      <th>C-E4-1</th>\n",
       "      <th>C-E4-2</th>\n",
       "      <th>C-F3-1</th>\n",
       "      <th>C-F3-2</th>\n",
       "      <th>C-F4-1</th>\n",
       "      <th>C-F4-2</th>\n",
       "      <th>C-G3-1</th>\n",
       "      <th>C-G3-2</th>\n",
       "      <th>C-G4-1</th>\n",
       "      <th>C-G4-2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C-A4-1</th>\n",
       "      <td>74</td>\n",
       "      <td>72</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-A4-2</th>\n",
       "      <td>71</td>\n",
       "      <td>74</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-B3-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>59</td>\n",
       "      <td>34</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-B3-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34</td>\n",
       "      <td>56</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>34</td>\n",
       "      <td>35</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-B4-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>33</td>\n",
       "      <td>28</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-B4-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>46</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-C3-1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>62</td>\n",
       "      <td>41</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-C3-2</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>41</td>\n",
       "      <td>79</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-C4-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>42</td>\n",
       "      <td>32</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-C4-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>32</td>\n",
       "      <td>49</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-D3-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36</td>\n",
       "      <td>34</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>58</td>\n",
       "      <td>35</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-D3-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36</td>\n",
       "      <td>35</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>35</td>\n",
       "      <td>57</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-D4-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>56</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-D4-2</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>59</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>25</td>\n",
       "      <td>27</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-E3-1</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "      <td>25</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>59</td>\n",
       "      <td>24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-E3-2</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "      <td>62</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-E4-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>58</td>\n",
       "      <td>17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-E4-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>62</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-F3-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71</td>\n",
       "      <td>31</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-F3-2</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31</td>\n",
       "      <td>75</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-F4-1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>54</td>\n",
       "      <td>48</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-F4-2</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>28</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>48</td>\n",
       "      <td>66</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-G3-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>60</td>\n",
       "      <td>53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-G3-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>53</td>\n",
       "      <td>57</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-G4-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C-G4-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       C-A4-1 C-A4-2 C-B3-1 C-B3-2 C-B4-1 C-B4-2 C-C3-1 C-C3-2 C-C4-1 C-C4-2  \\\n",
       "C-A4-1     74     72    NaN    NaN    NaN    NaN      1      3    NaN    NaN   \n",
       "C-A4-2     71     74    NaN    NaN    NaN    NaN      1      2    NaN    NaN   \n",
       "C-B3-1    NaN    NaN     59     34      1      1      1      1      1      2   \n",
       "C-B3-2    NaN    NaN     34     56      1      1      1      1      1      1   \n",
       "C-B4-1    NaN    NaN      1      1     33     28      1      1     12     12   \n",
       "C-B4-2    NaN    NaN      1      1     28     46      1      1     13     13   \n",
       "C-C3-1      1      1      1      1      1      1     62     41      1      1   \n",
       "C-C3-2      3      3      1      1      1      1     41     79      1      1   \n",
       "C-C4-1    NaN    NaN      1      1     12     13      1      1     42     32   \n",
       "C-C4-2    NaN    NaN      2      1     12     13      1      1     32     49   \n",
       "C-D3-1    NaN    NaN     36     34      1      1      1      1      1      1   \n",
       "C-D3-2    NaN    NaN     36     35      1      1      1      1      1      2   \n",
       "C-D4-1    NaN    NaN      2      2      1      1      1      1      1      1   \n",
       "C-D4-2      2      2      2      2    NaN    NaN      1      1      1      1   \n",
       "C-E3-1      2      2      1      1      1      1     24     25      1      1   \n",
       "C-E3-2      1      1      1      1      1      1     25     25      1      1   \n",
       "C-E4-1    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   \n",
       "C-E4-2    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   \n",
       "C-F3-1    NaN    NaN      2      1    NaN    NaN      1      2      2      2   \n",
       "C-F3-2      1      1      2      1    NaN    NaN      1      2    NaN    NaN   \n",
       "C-F4-1      1      1      3      3    NaN    NaN      1      1      2      2   \n",
       "C-F4-2      1      1      3      3    NaN    NaN      1      1      2      2   \n",
       "C-G3-1    NaN    NaN      2      1    NaN    NaN      1      1    NaN    NaN   \n",
       "C-G3-2    NaN    NaN      2      1    NaN    NaN      1      1    NaN    NaN   \n",
       "C-G4-1    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   \n",
       "C-G4-2    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   \n",
       "\n",
       "       C-D3-1 C-D3-2 C-D4-1 C-D4-2 C-E3-1 C-E3-2 C-E4-1 C-E4-2 C-F3-1 C-F3-2  \\\n",
       "C-A4-1    NaN    NaN    NaN      2      3      1    NaN    NaN    NaN      1   \n",
       "C-A4-2    NaN    NaN    NaN      2      2      1    NaN    NaN    NaN      1   \n",
       "C-B3-1     36     36      2      2      1      1    NaN    NaN      2      2   \n",
       "C-B3-2     34     35      2      2      1      1    NaN    NaN      1      1   \n",
       "C-B4-1      1      1      1    NaN      1      1    NaN    NaN    NaN    NaN   \n",
       "C-B4-2      1      1      1    NaN      1      1    NaN    NaN    NaN    NaN   \n",
       "C-C3-1      1      1      1      1     24     25    NaN    NaN      1      1   \n",
       "C-C3-2      1      1      1      1     26     25    NaN    NaN      2      2   \n",
       "C-C4-1      1      1      1      1      1      1    NaN    NaN      1    NaN   \n",
       "C-C4-2      1      2      1      1      1      1    NaN    NaN      1    NaN   \n",
       "C-D3-1     58     35      2      2      1      1    NaN    NaN      2      1   \n",
       "C-D3-2     35     57      2      2      1      1    NaN    NaN      1      1   \n",
       "C-D4-1      2      2     56     12      1      1      1      1      1      1   \n",
       "C-D4-2      2      2     12     59      1      1      1      2      2      1   \n",
       "C-E3-1      1      1      1      1     59     24    NaN    NaN      1      1   \n",
       "C-E3-2      1      1      1      1     24     62    NaN    NaN      1      1   \n",
       "C-E4-1    NaN    NaN      1      1    NaN    NaN     58     17    NaN    NaN   \n",
       "C-E4-2    NaN    NaN      1      2    NaN    NaN     17     62    NaN    NaN   \n",
       "C-F3-1      2      1      1      3      1      1    NaN    NaN     71     31   \n",
       "C-F3-2      1      1      1      1      1      1    NaN    NaN     31     75   \n",
       "C-F4-1      3      3     10     26      1      1      1      1      3      1   \n",
       "C-F4-2      3      3     11     28      1      1      1      1      3      1   \n",
       "C-G3-1      2      1      1      1      1      1    NaN    NaN     17     16   \n",
       "C-G3-2      2      1      1      1      2      1    NaN    NaN     17     16   \n",
       "C-G4-1    NaN    NaN      1      1    NaN    NaN     17     17    NaN    NaN   \n",
       "C-G4-2    NaN      1      1      1    NaN    NaN     21     17    NaN    NaN   \n",
       "\n",
       "       C-F4-1 C-F4-2 C-G3-1 C-G3-2 C-G4-1 C-G4-2  \n",
       "C-A4-1      1      1    NaN    NaN    NaN    NaN  \n",
       "C-A4-2      1      1    NaN    NaN    NaN    NaN  \n",
       "C-B3-1      3      3      2      2    NaN    NaN  \n",
       "C-B3-2      3      3      1      1    NaN    NaN  \n",
       "C-B4-1    NaN    NaN    NaN    NaN    NaN    NaN  \n",
       "C-B4-2    NaN    NaN    NaN    NaN    NaN    NaN  \n",
       "C-C3-1      1      1      1      1    NaN    NaN  \n",
       "C-C3-2      1      1      1      1    NaN    NaN  \n",
       "C-C4-1      1      1    NaN    NaN    NaN    NaN  \n",
       "C-C4-2      1      1    NaN    NaN    NaN    NaN  \n",
       "C-D3-1      3      3      2      2    NaN    NaN  \n",
       "C-D3-2      3      3      1      1    NaN      1  \n",
       "C-D4-1     10     11      1      1      1      1  \n",
       "C-D4-2     25     27      1      1      1      1  \n",
       "C-E3-1      1      1      1      2    NaN    NaN  \n",
       "C-E3-2      1      1      1      1    NaN    NaN  \n",
       "C-E4-1      1      1    NaN    NaN     17     21  \n",
       "C-E4-2      1      1    NaN    NaN     17     17  \n",
       "C-F3-1      3      3     17     17    NaN    NaN  \n",
       "C-F3-2      1      1     16     16    NaN    NaN  \n",
       "C-F4-1     54     48      1      1      1      1  \n",
       "C-F4-2     48     66      1      1      1      1  \n",
       "C-G3-1      1      1     60     53    NaN    NaN  \n",
       "C-G3-2      1      1     53     57    NaN    NaN  \n",
       "C-G4-1      1      1    NaN    NaN     17     17  \n",
       "C-G4-2      1      1    NaN    NaN     17     47  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create a dataframe containing all UIDs for each sample\n",
    "i = 0\n",
    "all_Emutations = pd.DataFrame()\n",
    "for sample in EmvcNEW.lineage.unique():\n",
    "    Emutations = pd.DataFrame()\n",
    "    Emutations = pd.DataFrame(EmvcNEW[EmvcNEW.lineage == sample].UID)\n",
    "    Emutations = Emutations.set_index(pd.Series(range(len(Emutations))))\n",
    "    Emutations.columns = [sample]\n",
    "    all_Emutations = pd.concat([all_Emutations,Emutations],axis=1)\n",
    "    i += 1\n",
    "    \n",
    "i = 0\n",
    "all_Pmutations = pd.DataFrame()\n",
    "for sample in PmvcNEW.lineage.unique():\n",
    "    Pmutations = pd.DataFrame()\n",
    "    Pmutations = pd.DataFrame(PmvcNEW[PmvcNEW.lineage == sample].UID)\n",
    "    Pmutations = Pmutations.set_index(pd.Series(range(len(Pmutations))))\n",
    "    Pmutations.columns = [sample]\n",
    "    all_Pmutations = pd.concat([all_Pmutations,Pmutations],axis=1)\n",
    "    i += 1\n",
    "    \n",
    "i = 0\n",
    "all_Cmutations = pd.DataFrame()\n",
    "for sample in CmvcNEW.lineage.unique():\n",
    "    Cmutations = pd.DataFrame()\n",
    "    Cmutations = pd.DataFrame(CmvcNEW[CmvcNEW.lineage == sample].UID)\n",
    "    Cmutations = Cmutations.set_index(pd.Series(range(len(Cmutations))))\n",
    "    Cmutations.columns = [sample]\n",
    "    all_Cmutations = pd.concat([all_Cmutations,Cmutations],axis=1)\n",
    "    i += 1\n",
    "\n",
    "i = 0\n",
    "all_Fmutations = pd.DataFrame()\n",
    "for sample in FmvcNEW.lineage.unique():\n",
    "    Fmutations = pd.DataFrame()\n",
    "    Fmutations = pd.DataFrame(FmvcNEW[FmvcNEW.lineage == sample].UID)\n",
    "    Fmutations = Fmutations.set_index(pd.Series(range(len(Fmutations))))\n",
    "    Fmutations.columns = [sample]\n",
    "    all_Fmutations = pd.concat([all_Fmutations,Fmutations],axis=1)\n",
    "    i += 1\n",
    "\n",
    "# create a comparative dataframe that contains the number of shared UIDs for each sample pair\n",
    "compare_Emutations = pd.DataFrame(index=all_Emutations.columns,columns=all_Emutations.columns)\n",
    "for column in compare_Emutations.columns:\n",
    "    for row in compare_Emutations.columns:\n",
    "        testrow = all_Emutations[row].dropna()\n",
    "        testcolumn = all_Emutations[column].dropna()\n",
    "        test = testrow.isin(testcolumn)\n",
    "        count = test.value_counts()\n",
    "        if True in count:\n",
    "            compare_Emutations[column][row] = count[True]\n",
    "\n",
    "compare_Pmutations = pd.DataFrame(index=all_Pmutations.columns,columns=all_Pmutations.columns)\n",
    "for column in compare_Pmutations.columns:\n",
    "    for row in compare_Pmutations.columns:\n",
    "        testrow = all_Pmutations[row].dropna()\n",
    "        testcolumn = all_Pmutations[column].dropna()\n",
    "        test = testrow.isin(testcolumn)\n",
    "        count = test.value_counts()\n",
    "        if True in count:\n",
    "            compare_Pmutations[column][row] = count[True]\n",
    "\n",
    "compare_Cmutations = pd.DataFrame(index=all_Cmutations.columns,columns=all_Cmutations.columns)\n",
    "for column in compare_Cmutations.columns:\n",
    "    for row in compare_Cmutations.columns:\n",
    "        testrow = all_Cmutations[row].dropna()\n",
    "        testcolumn = all_Cmutations[column].dropna()\n",
    "        test = testrow.isin(testcolumn)\n",
    "        count = test.value_counts()\n",
    "        if True in count:\n",
    "            compare_Cmutations[column][row] = count[True]\n",
    "\n",
    "compare_Fmutations = pd.DataFrame(index=all_Fmutations.columns,columns=all_Fmutations.columns)\n",
    "for column in compare_Fmutations.columns:\n",
    "    for row in compare_Fmutations.columns:\n",
    "        testrow = all_Fmutations[row].dropna()\n",
    "        testcolumn = all_Fmutations[column].dropna()\n",
    "        test = testrow.isin(testcolumn)\n",
    "        count = test.value_counts()\n",
    "        if True in count:\n",
    "            compare_Fmutations[column][row] = count[True]\n",
    "       \n",
    "compare_Emutations = compare_Emutations.sort_index()\n",
    "compare_Emutations = compare_Emutations.sort_index(axis = 1)\n",
    "compare_Pmutations = compare_Pmutations.sort_index()\n",
    "compare_Pmutations = compare_Pmutations.sort_index(axis = 1)\n",
    "compare_Cmutations = compare_Cmutations.sort_index()\n",
    "compare_Cmutations = compare_Cmutations.sort_index(axis = 1)\n",
    "compare_Fmutations = compare_Fmutations.sort_index()\n",
    "compare_Fmutations = compare_Fmutations.sort_index(axis = 1)\n",
    "compare_Cmutations\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Begin to analyze what's going on with some of the very frequently called hits. [positionS FROM MG1655v3]\n",
    "Sort by UID. UIDs hit very frequently in patterns across the population may have been present in the founder lines.\n",
    "\n",
    "ECNR2.1 seems to especially have a few founder mutations, and interestingly well A1, uniquely, does not share the two most abundant mutations. Those two mutations are present in every other evolved line, but not the founder line, and are as follows: POS 1287826 - Gene purU AND POS 573440 - Gene rusA. Then the third most abundant mutation is additionally not shared by well G1, and only present in one of two colonies from well F2, and is as follows: POS 4351093 - Gene dcuS. Then the fourth most abundant mutation is additionally not shared by wells C2 and H2: POS 2248780 - Gene yeiE.\n",
    "\n",
    "P populations have some variants that appear to be founders: POS 2288554, 53006, 2995224, 1139782, 1711495, 588492, 4585812, 2614611, 1423525, 2741231, 3309007, 4566573, 1995714, 66624, 1946418, 49764, 1873312, 384055, 2416764\n",
    "\n",
    "Discount founder mutations from population branch points.\n",
    "\n",
    "F population groups:  \n",
    "wells E5/E6/F5/G5/G6  \n",
    "wells B5/B6/C6/H5  \n",
    "  \n",
    "C population groups:  \n",
    "wells B3/D3  \n",
    "wells B4/C4  \n",
    "wells C3/E3  \n",
    "wells D4/F4  \n",
    "wells E4/G4  \n",
    "wells F3/G3  \n",
    "  \n",
    "P populatin groups:  \n",
    "wells 5/6/11  \n",
    "wells 5/6  \n",
    "wells 8/13\n",
    "  \n",
    "Poor read quality alleles found in all four populations: POS 1594124 and 1594248 - Gene yneO, POS 380012 - Gene yaiO, POS 380012 - Gene mtr, POS 1050297 and 1050308 in gene insB1 (a heavily mutated transposable element), "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Culled gene variant UIDs that occur most frequently include: \n",
      "           ECNR2  C321  C321_dA  C321_dA_F11\n",
      "245949d6   17.0   7.0      8.0          4.0\n",
      "9bbad34b   11.0   6.0      1.0          6.0\n",
      "b7b65239   10.0   NaN      2.0          9.0\n",
      "1e2dcc3d    8.0   NaN      NaN          NaN\n",
      "fde7e5b8    8.0   NaN      NaN          7.0\n",
      "2ee2dd00    7.0   NaN      NaN          NaN\n",
      "54a0e7ce    7.0   1.0      9.0          4.0\n",
      "73812a16    7.0   2.0      2.0          8.0\n",
      "0ee8f534    6.0   NaN      NaN          NaN\n",
      "17131e16    6.0   NaN      NaN          NaN\n",
      "5f36ac54    6.0   NaN      NaN          4.0\n",
      "b062d7f8    6.0   5.0      6.0          NaN\n",
      "2953bf45    2.0   5.0      NaN          6.0\n",
      "206b7322    NaN   8.0      NaN          NaN\n",
      "9b1974cd    NaN   NaN      5.0         14.0\n",
      "e5e8bfc2    NaN   7.0      4.0          2.0\n",
      "f1e9f70d    NaN   NaN      2.0          6.0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>P-1-1</th>\n",
       "      <th>P-1-2</th>\n",
       "      <th>P-10-1</th>\n",
       "      <th>P-10-2</th>\n",
       "      <th>P-11-1</th>\n",
       "      <th>P-11-2</th>\n",
       "      <th>P-12-1</th>\n",
       "      <th>P-12-2</th>\n",
       "      <th>P-13-1</th>\n",
       "      <th>P-14-1</th>\n",
       "      <th>P-14-2</th>\n",
       "      <th>P-2-1</th>\n",
       "      <th>P-2-2</th>\n",
       "      <th>P-3-1</th>\n",
       "      <th>P-3-2</th>\n",
       "      <th>P-4-1</th>\n",
       "      <th>P-4-2</th>\n",
       "      <th>P-5-1</th>\n",
       "      <th>P-5-2</th>\n",
       "      <th>P-6-1</th>\n",
       "      <th>P-6-2</th>\n",
       "      <th>P-7-1</th>\n",
       "      <th>P-7-2</th>\n",
       "      <th>P-8-1</th>\n",
       "      <th>P-8-2</th>\n",
       "      <th>P-9-1</th>\n",
       "      <th>P-9-2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P-1-1</th>\n",
       "      <td>53</td>\n",
       "      <td>49</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-1-2</th>\n",
       "      <td>49</td>\n",
       "      <td>53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-10-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>59</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-10-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31</td>\n",
       "      <td>52</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-11-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>51</td>\n",
       "      <td>26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-11-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>49</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-12-1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>70</td>\n",
       "      <td>41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-12-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>41</td>\n",
       "      <td>82</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-13-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-14-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>160</td>\n",
       "      <td>70</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-14-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>69</td>\n",
       "      <td>116</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-2-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>75</td>\n",
       "      <td>51</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-2-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>51</td>\n",
       "      <td>62</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-3-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-3-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-4-1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>54</td>\n",
       "      <td>27</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-4-2</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27</td>\n",
       "      <td>56</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-5-1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>75</td>\n",
       "      <td>67</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-5-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>67</td>\n",
       "      <td>72</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-6-1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>73</td>\n",
       "      <td>55</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-6-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>55</td>\n",
       "      <td>57</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-7-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-7-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>45</td>\n",
       "      <td>64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-8-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36</td>\n",
       "      <td>35</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-8-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-9-1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>97</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-9-2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>61</td>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       P-1-1 P-1-2 P-10-1 P-10-2 P-11-1 P-11-2 P-12-1 P-12-2 P-13-1 P-14-1  \\\n",
       "P-1-1     53    49    NaN    NaN    NaN    NaN      1    NaN    NaN    NaN   \n",
       "P-1-2     49    53    NaN    NaN    NaN    NaN      1    NaN    NaN      1   \n",
       "P-10-1   NaN   NaN     59     31      1      1    NaN    NaN    NaN    NaN   \n",
       "P-10-2   NaN   NaN     31     52      1      1    NaN    NaN    NaN    NaN   \n",
       "P-11-1   NaN   NaN      1      1     51     26    NaN    NaN    NaN    NaN   \n",
       "P-11-2   NaN   NaN      1      1     26     49    NaN      1    NaN    NaN   \n",
       "P-12-1     1     1    NaN    NaN    NaN    NaN     70     41    NaN    NaN   \n",
       "P-12-2   NaN   NaN    NaN    NaN    NaN      1     41     82    NaN    NaN   \n",
       "P-13-1   NaN   NaN    NaN    NaN    NaN    NaN    NaN    NaN      5      1   \n",
       "P-14-1   NaN     1    NaN    NaN    NaN    NaN    NaN    NaN      1    160   \n",
       "P-14-2   NaN   NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN     69   \n",
       "P-2-1    NaN   NaN      1      1      2      2    NaN    NaN    NaN      1   \n",
       "P-2-2    NaN   NaN      1      1      2      2      1    NaN    NaN      1   \n",
       "P-3-1    NaN   NaN      2      2    NaN    NaN      1      1    NaN    NaN   \n",
       "P-3-2    NaN   NaN      1      1    NaN    NaN    NaN    NaN    NaN    NaN   \n",
       "P-4-1      1     1      1      1      1      1      1    NaN    NaN    NaN   \n",
       "P-4-2      1     1      1      1      1      1      1    NaN    NaN    NaN   \n",
       "P-5-1      1     1    NaN    NaN    NaN    NaN      1    NaN    NaN    NaN   \n",
       "P-5-2    NaN   NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   \n",
       "P-6-1      1     1    NaN    NaN      1      1      1    NaN    NaN    NaN   \n",
       "P-6-2    NaN   NaN    NaN    NaN      1      1    NaN    NaN    NaN    NaN   \n",
       "P-7-1    NaN     1      2      2    NaN    NaN      1      1    NaN      1   \n",
       "P-7-2    NaN     1      2      2    NaN    NaN      1      1    NaN      1   \n",
       "P-8-1    NaN   NaN    NaN    NaN    NaN    NaN    NaN    NaN      4    NaN   \n",
       "P-8-2    NaN   NaN    NaN    NaN    NaN    NaN    NaN    NaN      4    NaN   \n",
       "P-9-1    NaN   NaN      1      1      2    NaN    NaN      1    NaN    NaN   \n",
       "P-9-2    NaN   NaN      1      1    NaN    NaN    NaN      1    NaN    NaN   \n",
       "\n",
       "       P-14-2 P-2-1 P-2-2 P-3-1 P-3-2 P-4-1 P-4-2 P-5-1 P-5-2 P-6-1 P-6-2  \\\n",
       "P-1-1     NaN   NaN   NaN   NaN   NaN     1     1     1   NaN     1   NaN   \n",
       "P-1-2     NaN   NaN   NaN   NaN   NaN     1     1     1   NaN     1   NaN   \n",
       "P-10-1    NaN     1     1     2     1     1     1   NaN   NaN   NaN   NaN   \n",
       "P-10-2    NaN     1     1     2     1     1     1   NaN   NaN   NaN   NaN   \n",
       "P-11-1    NaN     2     2   NaN   NaN     1     1   NaN   NaN     1     1   \n",
       "P-11-2    NaN     2     2   NaN   NaN     1     1   NaN   NaN     1     1   \n",
       "P-12-1    NaN   NaN     1     1   NaN     1     1     1   NaN     1   NaN   \n",
       "P-12-2    NaN   NaN   NaN     1   NaN   NaN   NaN   NaN   NaN   NaN   NaN   \n",
       "P-13-1    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   \n",
       "P-14-1     70     1     1   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   \n",
       "P-14-2    116     1     1   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   \n",
       "P-2-1       1    75    51   NaN   NaN     1     1   NaN   NaN     1   NaN   \n",
       "P-2-2       1    51    62   NaN   NaN     1     1   NaN   NaN   NaN   NaN   \n",
       "P-3-1     NaN   NaN   NaN    38    10     1   NaN   NaN   NaN   NaN   NaN   \n",
       "P-3-2     NaN   NaN   NaN    10    16     1   NaN   NaN   NaN   NaN   NaN   \n",
       "P-4-1     NaN     1     1     1     1    54    27     1   NaN     1   NaN   \n",
       "P-4-2     NaN     1     1   NaN   NaN    27    56     1   NaN     1   NaN   \n",
       "P-5-1     NaN   NaN   NaN   NaN   NaN     1     1    75    67     6   NaN   \n",
       "P-5-2     NaN   NaN   NaN   NaN   NaN   NaN   NaN    67    72   NaN   NaN   \n",
       "P-6-1     NaN     1   NaN   NaN   NaN     1     1     6   NaN    73    55   \n",
       "P-6-2     NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN    55    57   \n",
       "P-7-1     NaN   NaN   NaN     3     1   NaN   NaN   NaN   NaN   NaN   NaN   \n",
       "P-7-2     NaN   NaN   NaN     3     1     1     1   NaN   NaN   NaN   NaN   \n",
       "P-8-1     NaN     1     1   NaN   NaN   NaN   NaN     1   NaN     1   NaN   \n",
       "P-8-2     NaN     1     1   NaN   NaN   NaN   NaN     1   NaN     1   NaN   \n",
       "P-9-1     NaN     1     1     1   NaN   NaN   NaN   NaN   NaN   NaN   NaN   \n",
       "P-9-2     NaN   NaN   NaN     1   NaN   NaN   NaN   NaN   NaN   NaN   NaN   \n",
       "\n",
       "       P-7-1 P-7-2 P-8-1 P-8-2 P-9-1 P-9-2  \n",
       "P-1-1    NaN   NaN   NaN   NaN   NaN   NaN  \n",
       "P-1-2      1     1   NaN   NaN   NaN   NaN  \n",
       "P-10-1     2     2   NaN   NaN     1     1  \n",
       "P-10-2     2     2   NaN   NaN     1     1  \n",
       "P-11-1   NaN   NaN   NaN   NaN     2   NaN  \n",
       "P-11-2   NaN   NaN   NaN   NaN   NaN   NaN  \n",
       "P-12-1     1     1   NaN   NaN   NaN   NaN  \n",
       "P-12-2     1     1   NaN   NaN     1     1  \n",
       "P-13-1   NaN   NaN     4     4   NaN   NaN  \n",
       "P-14-1     1     1   NaN   NaN   NaN   NaN  \n",
       "P-14-2   NaN   NaN   NaN   NaN   NaN   NaN  \n",
       "P-2-1    NaN   NaN     1     1     1   NaN  \n",
       "P-2-2    NaN   NaN     1     1     1   NaN  \n",
       "P-3-1      3     3   NaN   NaN     1     1  \n",
       "P-3-2      1     1   NaN   NaN   NaN   NaN  \n",
       "P-4-1    NaN     1   NaN   NaN   NaN   NaN  \n",
       "P-4-2    NaN     1   NaN   NaN   NaN   NaN  \n",
       "P-5-1    NaN   NaN     1     1   NaN   NaN  \n",
       "P-5-2    NaN   NaN   NaN   NaN   NaN   NaN  \n",
       "P-6-1    NaN   NaN     1     1   NaN   NaN  \n",
       "P-6-2    NaN   NaN   NaN   NaN   NaN   NaN  \n",
       "P-7-1     67    45   NaN   NaN     1     1  \n",
       "P-7-2     45    64   NaN   NaN     1     1  \n",
       "P-8-1    NaN   NaN    36    35     1   NaN  \n",
       "P-8-2    NaN   NaN    35    35     1   NaN  \n",
       "P-9-1      1     1     1     1    97    61  \n",
       "P-9-2      1     1   NaN   NaN    61    87  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "founder_groups = [['F-E5-1','F-E5-2','F-E6-1','F-E6-2','F-F5-1','F-F5-2','F-G5-1','F-G5-2','F-G6-1','F-G6-2'],\n",
    "                  ['F-B5-1','F-B5-2','F-B6-1','F-B6-2','F-C6-1','F-C6-2','F-H5-1','F-H5-2'],\n",
    "                  ['C-B3-1','C-B3-2','C-D3-1','C-D3-2'],['C-B4-1','C-B4-2','C-C4-1','C-C4-2'],\n",
    "                  ['C-C3-1','C-C3-2','C-E3-1','C-E3-2'],['C-D4-1','C-D4-2','C-F4-1','C-F4-2'],\n",
    "                  ['C-E4-1','C-E4-2','C-G4-1','C-G4-2'],['C-F3-1','C-F3-2','C-G3-1','C-G3-2'],\n",
    "                  ['P-5-1','P-6-1','P-11-1','P-5-2','P-6-2','P-11-2'],\n",
    "                  ['P-5-1','P-6-1','P-5-2','P-6-2'],\n",
    "                  ['P-8-1','P-8-2','P-13-1','P-13-2']]\n",
    "\n",
    "# remove all but two variants from mutations that occur in all members of any founder group\n",
    "EmvcNEWc = EmvcNEW.copy()\n",
    "PmvcNEWc = PmvcNEW.copy()\n",
    "CmvcNEWc = CmvcNEW.copy()\n",
    "FmvcNEWc = FmvcNEW.copy()\n",
    "for g in founder_groups:\n",
    "    if g[0][0] == 'C':\n",
    "        for v in CmvcNEWc[CmvcNEWc.lineage == g[0]].UID.tolist():\n",
    "            present = True\n",
    "            for s in g:\n",
    "                if v not in CmvcNEWc[CmvcNEWc.lineage == s].UID.tolist():\n",
    "                    present = False\n",
    "            if present:\n",
    "                for s in g[2:]:\n",
    "                    CmvcNEWc = CmvcNEWc[~((CmvcNEWc.lineage == s) & (CmvcNEWc.UID == v))]\n",
    "    elif g[0][0] == 'F':\n",
    "        for v in FmvcNEWc[FmvcNEWc.lineage == g[0]].UID.tolist():\n",
    "            present = True\n",
    "            for s in g:\n",
    "                if v not in FmvcNEWc[FmvcNEWc.lineage == s].UID.tolist():\n",
    "                    present = False\n",
    "            if present:\n",
    "                for s in g[2:]:\n",
    "                    FmvcNEWc = FmvcNEWc[~((FmvcNEWc.lineage == s) & (FmvcNEWc.UID == v))]\n",
    "    elif g[0][0] == 'P':\n",
    "        for v in PmvcNEWc[PmvcNEWc.lineage == g[0]].UID.tolist():\n",
    "            present = True\n",
    "            for s in g:\n",
    "                if v not in PmvcNEWc[PmvcNEWc.lineage == s].UID.tolist():\n",
    "                    present = False\n",
    "            if present:\n",
    "                for s in g[2:]:\n",
    "                    PmvcNEWc = PmvcNEWc[~((PmvcNEWc.lineage == s) & (PmvcNEWc.UID == v))]\n",
    "            \n",
    "\n",
    "problem_POS = [1287826, 573440, 4351093, 2248780, 1594124, 1594248, 1050297, 1050308, 380012, 3304806, 2288554, 53006, \n",
    "               2995224, 1139782, 1711495, 588492, 4585812, 2614611, 1423525, 2741231, 3309007, 4566573, 1995714, 66624, \n",
    "               1946418, 49764, 1873312, 384055, 2416764]\n",
    "\n",
    "# remove problem POS from variant lists\n",
    "EmvcNEWc = EmvcNEWc[~EmvcNEWc.position.isin(problem_POS)]\n",
    "PmvcNEWc = PmvcNEWc[~PmvcNEWc.position.isin(problem_POS)]\n",
    "CmvcNEWc = CmvcNEWc[~CmvcNEWc.position.isin(problem_POS)]\n",
    "FmvcNEWc = FmvcNEWc[~FmvcNEWc.position.isin(problem_POS)]\n",
    "\n",
    "# create a dataframe containing all UIDs for each sample\n",
    "i = 0\n",
    "all_EmutationsC = pd.DataFrame()\n",
    "for sample in EmvcNEWc.lineage.unique():\n",
    "    Emutations = pd.DataFrame()\n",
    "    Emutations = pd.DataFrame(EmvcNEWc[EmvcNEWc.lineage == sample].UID)\n",
    "    Emutations = Emutations.set_index(pd.Series(range(len(Emutations))))\n",
    "    Emutations.columns = [sample]\n",
    "    all_EmutationsC = pd.concat([all_EmutationsC,Emutations],axis=1)\n",
    "    i += 1\n",
    "\n",
    "i = 0\n",
    "all_PmutationsC = pd.DataFrame()\n",
    "for sample in PmvcNEWc.lineage.unique():\n",
    "    Pmutations = pd.DataFrame()\n",
    "    Pmutations = pd.DataFrame(PmvcNEWc[PmvcNEWc.lineage == sample].UID)\n",
    "    Pmutations = Pmutations.set_index(pd.Series(range(len(Pmutations))))\n",
    "    Pmutations.columns = [sample]\n",
    "    all_PmutationsC = pd.concat([all_PmutationsC,Pmutations],axis=1)\n",
    "    i += 1\n",
    "    \n",
    "i = 0\n",
    "all_CmutationsC = pd.DataFrame()\n",
    "for sample in CmvcNEWc.lineage.unique():\n",
    "    Cmutations = pd.DataFrame()\n",
    "    Cmutations = pd.DataFrame(CmvcNEWc[CmvcNEWc.lineage == sample].UID)\n",
    "    Cmutations = Cmutations.set_index(pd.Series(range(len(Cmutations))))\n",
    "    Cmutations.columns = [sample]\n",
    "    all_CmutationsC = pd.concat([all_CmutationsC,Cmutations],axis=1)\n",
    "    i += 1\n",
    "\n",
    "i = 0\n",
    "all_FmutationsC = pd.DataFrame()\n",
    "for sample in FmvcNEWc.lineage.unique():\n",
    "    Fmutations = pd.DataFrame()\n",
    "    Fmutations = pd.DataFrame(FmvcNEWc[FmvcNEWc.lineage == sample].UID)\n",
    "    Fmutations = Fmutations.set_index(pd.Series(range(len(Fmutations))))\n",
    "    Fmutations.columns = [sample]\n",
    "    all_FmutationsC = pd.concat([all_FmutationsC,Fmutations],axis=1)\n",
    "    i += 1\n",
    "\n",
    "# create a comparative dataframe that contains the number of shared UIDs for each sample pair\n",
    "compare_EmutationsC = pd.DataFrame(index=all_EmutationsC.columns,columns=all_EmutationsC.columns)\n",
    "for column in compare_EmutationsC.columns:\n",
    "    for row in compare_EmutationsC.columns:\n",
    "        testrow = all_EmutationsC[row].dropna()\n",
    "        testcolumn = all_EmutationsC[column].dropna()\n",
    "        test = testrow.isin(testcolumn)\n",
    "        count = test.value_counts()\n",
    "        if True in count:\n",
    "            compare_EmutationsC[column][row] = count[True]\n",
    "\n",
    "compare_PmutationsC = pd.DataFrame(index=all_PmutationsC.columns,columns=all_PmutationsC.columns)\n",
    "for column in compare_PmutationsC.columns:\n",
    "    for row in compare_PmutationsC.columns:\n",
    "        testrow = all_PmutationsC[row].dropna()\n",
    "        testcolumn = all_PmutationsC[column].dropna()\n",
    "        test = testrow.isin(testcolumn)\n",
    "        count = test.value_counts()\n",
    "        if True in count:\n",
    "            compare_PmutationsC[column][row] = count[True]\n",
    "\n",
    "compare_CmutationsC = pd.DataFrame(index=all_CmutationsC.columns,columns=all_CmutationsC.columns)\n",
    "for column in compare_CmutationsC.columns:\n",
    "    for row in compare_CmutationsC.columns:\n",
    "        testrow = all_CmutationsC[row].dropna()\n",
    "        testcolumn = all_CmutationsC[column].dropna()\n",
    "        test = testrow.isin(testcolumn)\n",
    "        count = test.value_counts()\n",
    "        if True in count:\n",
    "            compare_CmutationsC[column][row] = count[True]\n",
    "\n",
    "compare_FmutationsC = pd.DataFrame(index=all_FmutationsC.columns,columns=all_FmutationsC.columns)\n",
    "for column in compare_FmutationsC.columns:\n",
    "    for row in compare_FmutationsC.columns:\n",
    "        testrow = all_FmutationsC[row].dropna()\n",
    "        testcolumn = all_FmutationsC[column].dropna()\n",
    "        test = testrow.isin(testcolumn)\n",
    "        count = test.value_counts()\n",
    "        if True in count:\n",
    "            compare_FmutationsC[column][row] = count[True]\n",
    "       \n",
    "compare_EmutationsC = compare_EmutationsC.sort_index()\n",
    "compare_EmutationsC = compare_EmutationsC.sort_index(axis = 1)\n",
    "compare_PmutationsC = compare_PmutationsC.sort_index()\n",
    "compare_PmutationsC = compare_PmutationsC.sort_index(axis = 1)\n",
    "compare_CmutationsC = compare_CmutationsC.sort_index()\n",
    "compare_CmutationsC = compare_CmutationsC.sort_index(axis = 1)\n",
    "compare_FmutationsC = compare_FmutationsC.sort_index()\n",
    "compare_FmutationsC = compare_FmutationsC.sort_index(axis = 1)\n",
    "\n",
    "all_UIDs = pd.concat([all_EmutationsC.stack().value_counts(),all_PmutationsC.stack().value_counts(),\n",
    "           all_CmutationsC.stack().value_counts(),all_FmutationsC.stack().value_counts()],axis = 1)\n",
    "all_UIDs.columns = ['ECNR2','C321','C321_dA','C321_dA_F11']\n",
    "print(\"Culled gene variant UIDs that occur most frequently include: \\n\",\n",
    "      all_UIDs[(all_UIDs.ECNR2 > 5) | (all_UIDs.C321 > 5) | (all_UIDs.C321_dA > 5) | (all_UIDs.C321_dA_F11 > 5)].sort_values(by = 'ECNR2',ascending = False))\n",
    "\n",
    "\n",
    "compare_PmutationsC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "number of genes hit at or over threshold in evolved \n",
      "ECNR2.1 populations: 15 \n",
      "C321 poipulations: 26 \n",
      "C321.∆A poipulations: 18 \n",
      "C321.∆A_F11 populations: 14 \n",
      "\n",
      "genes hit: \n",
      "       ECNR2  C321  C321_dA  C321_dA_F11\n",
      "aceE    NaN   1.0      7.0          2.0\n",
      "aceF    1.0   5.0      NaN          NaN\n",
      "astD    NaN   5.0      NaN          NaN\n",
      "fdnG    NaN   NaN      5.0          NaN\n",
      "fimH    1.0   2.0      3.0         19.0\n",
      "fis     NaN   7.0     11.0          NaN\n",
      "flu    31.0   NaN      9.0         22.0\n",
      "folA    NaN  20.0     10.0          4.0\n",
      "folM    NaN   8.0      1.0          NaN\n",
      "glrR    NaN   6.0      NaN          NaN\n",
      "gltB    NaN   5.0      4.0          NaN\n",
      "kup     1.0   5.0      3.0          NaN\n",
      "leuP    7.0   1.0      9.0          4.0\n",
      "leuQ    2.0   NaN      3.0          5.0\n",
      "leuV    8.0   NaN      NaN          7.0\n",
      "malK    NaN   NaN      5.0          2.0\n",
      "mdtJ    NaN   NaN      2.0          6.0\n",
      "mutS    NaN   5.0      NaN          NaN\n",
      "narX    5.0   NaN      2.0          NaN\n",
      "nfrA    NaN   5.0      NaN          NaN\n",
      "nusA    NaN   6.0      2.0          NaN\n",
      "ompT   10.0   2.0      NaN          1.0\n",
      "oxyR    NaN   NaN      7.0         22.0\n",
      "phoP    6.0   NaN      1.0          9.0\n",
      "prfB    2.0   NaN     10.0         14.0\n",
      "prfC    NaN   NaN      3.0          5.0\n",
      "pta     NaN  10.0      1.0          NaN\n",
      "purL    2.0   6.0      2.0          NaN\n",
      "purT    NaN   6.0      2.0         14.0\n",
      "pykF   14.0   7.0      1.0          3.0\n",
      "pyrC   10.0   2.0      NaN          NaN\n",
      "pyrE   30.0  13.0     11.0         14.0\n",
      "recE    6.0   2.0      NaN          3.0\n",
      "rlmB    3.0   5.0      2.0          6.0\n",
      "rph     2.0   6.0      4.0         16.0\n",
      "rpoB   13.0   9.0      3.0          NaN\n",
      "rpoC   18.0  14.0      6.0          3.0\n",
      "rsxC    1.0   7.0      NaN          2.0\n",
      "thiD    5.0   NaN      2.0          NaN\n",
      "yahL    NaN   NaN      5.0          2.0\n",
      "yccS    4.0   7.0      3.0          NaN\n",
      "ycgB    NaN   1.0      6.0          NaN\n",
      "ydfI    5.0   2.0      NaN          NaN\n",
      "ydfJ    1.0   NaN      5.0          2.0\n",
      "yeeJ    NaN   NaN      7.0          4.0\n",
      "yehB    1.0   NaN      7.0          NaN\n",
      "yfcU    NaN   5.0      2.0          NaN\n",
      "yfhM    3.0   6.0      1.0          1.0\n",
      "yfiH    6.0   5.0      6.0          NaN\n",
      "ygeR    NaN   NaN      1.0          5.0\n",
      "yhjK    NaN   1.0      5.0          NaN\n",
      "yneO    NaN   7.0      2.0          NaN\n",
      "\n",
      "intergenic hits: \n",
      "          ECNR2  C321  C321_dA  C321_dA_F11\n",
      "49766      NaN   7.0      4.0          2.0\n",
      "311472     NaN   4.0      1.0          NaN\n",
      "1212080    3.0   NaN      2.0          2.0\n",
      "1263627    NaN   NaN      3.0          NaN\n",
      "1351189    7.0   2.0      2.0          8.0\n",
      "1392950    NaN   3.0      NaN          NaN\n",
      "1818543    2.0   3.0      1.0          4.0\n",
      "2057669    1.0   NaN      3.0          1.0\n",
      "2071304    4.0   NaN      NaN          NaN\n",
      "2731346    NaN   3.0      NaN          NaN\n",
      "3931176    NaN   NaN      2.0          6.0\n"
     ]
    }
   ],
   "source": [
    "threshold_hitsC = pd.concat([EmvcNEWc.gene.value_counts(),PmvcNEWc.gene.value_counts(),\n",
    "                             CmvcNEWc.gene.value_counts(),FmvcNEWc.gene.value_counts()], axis=1)\n",
    "threshold_hitsC.columns = ['ECNR2','C321','C321_dA','C321_dA_F11']\n",
    "threshold_hitsC = threshold_hitsC[(threshold_hitsC.ECNR2 >= threshold) | (threshold_hitsC.C321 >= threshold) | \n",
    "                                  (threshold_hitsC.C321_dA >= threshold) | (threshold_hitsC.C321_dA_F11 >= threshold)]\n",
    "\n",
    "intergenic_hitsC = pd.concat([EmvcNEWc[EmvcNEWc.gene.isnull()].position.value_counts(),\n",
    "                              PmvcNEWc[PmvcNEWc.gene.isnull()].position.value_counts(),\n",
    "                              CmvcNEWc[CmvcNEWc.gene.isnull()].position.value_counts(),\n",
    "                              FmvcNEWc[FmvcNEWc.gene.isnull()].position.value_counts()], axis=1)\n",
    "intergenic_hitsC.columns = ['ECNR2','C321','C321_dA','C321_dA_F11']\n",
    "intergenic_hitsC = intergenic_hitsC[(intergenic_hitsC.ECNR2 >= positional_threshold) | \n",
    "                                    (intergenic_hitsC.C321 >= positional_threshold) | \n",
    "                                    (intergenic_hitsC.C321_dA >= positional_threshold) |\n",
    "                                    (intergenic_hitsC.C321_dA_F11 >= positional_threshold)]\n",
    "\n",
    "print('\\nnumber of genes hit at or over threshold in evolved \\nECNR2.1 populations:',\n",
    "      threshold_hitsC[threshold_hitsC.ECNR2 >= threshold].ECNR2.count(),\n",
    "      '\\nC321 poipulations:',\n",
    "      threshold_hitsC[threshold_hitsC.C321 >= threshold].C321.count(),\n",
    "      '\\nC321.∆A poipulations:',\n",
    "      threshold_hitsC[threshold_hitsC.C321_dA >= threshold].C321_dA.count(),\n",
    "      '\\nC321.∆A_F11 populations:',\n",
    "      threshold_hitsC[threshold_hitsC.C321_dA_F11 >= threshold].C321_dA_F11.count(),'\\n')\n",
    "\n",
    "writer = pd.ExcelWriter('gene_variants.xlsx', engine='xlsxwriter')\n",
    "ancestral_variants.to_excel(writer,sheet_name='ancestral')\n",
    "threshold_hits.to_excel(writer,sheet_name='variants_base')\n",
    "threshold_hitsC.to_excel(writer,sheet_name='variants_culled')\n",
    "writer.save()\n",
    "\n",
    "print('genes hit: \\n',threshold_hitsC)\n",
    "print('\\nintergenic hits: \\n',intergenic_hitsC)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "49766      4\n",
       "2057669    3\n",
       "1263627    3\n",
       "2168525    2\n",
       "4058263    2\n",
       "4094415    2\n",
       "3060764    2\n",
       "2498548    2\n",
       "84         2\n",
       "1212080    2\n",
       "1673789    2\n",
       "3915259    2\n",
       "2765411    2\n",
       "3700498    2\n",
       "1351189    2\n",
       "830697     2\n",
       "3931176    2\n",
       "3659139    2\n",
       "2877939    2\n",
       "1097138    2\n",
       "3234679    2\n",
       "4037496    2\n",
       "3367768    2\n",
       "1097030    2\n",
       "915240     1\n",
       "3774258    1\n",
       "3384481    1\n",
       "3991086    1\n",
       "3754521    1\n",
       "1151486    1\n",
       "107544     1\n",
       "3841840    1\n",
       "2819222    1\n",
       "311472     1\n",
       "2068207    1\n",
       "1502224    1\n",
       "755087     1\n",
       "1754636    1\n",
       "3828745    1\n",
       "780976     1\n",
       "1818543    1\n",
       "1310401    1\n",
       "4352433    1\n",
       "4153616    1\n",
       "2304635    1\n",
       "335221     1\n",
       "945780     1\n",
       "3925800    1\n",
       "841585     1\n",
       "401135     1\n",
       "3239660    1\n",
       "4037093    1\n",
       "1214175    1\n",
       "1270159    1\n",
       "4153595    1\n",
       "2229198    1\n",
       "1237323    1\n",
       "2788803    1\n",
       "1496700    1\n",
       "34111      1\n",
       "4168472    1\n",
       "2222128    1\n",
       "2718645    1\n",
       "4281699    1\n",
       "Name: position, dtype: int64"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CmvcNEWc[CmvcNEWc.gene.isnull()].position.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UID</th>\n",
       "      <th>position</th>\n",
       "      <th>lineage</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>DP</th>\n",
       "      <th>AF</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>430400</th>\n",
       "      <td>166a1092</td>\n",
       "      <td>3815879</td>\n",
       "      <td>C-G4-1</td>\n",
       "      <td>TCC</td>\n",
       "      <td>TCCC</td>\n",
       "      <td>82.0</td>\n",
       "      <td>0.961538</td>\n",
       "      <td>rph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430410</th>\n",
       "      <td>166a1092</td>\n",
       "      <td>3815879</td>\n",
       "      <td>C-G4-2</td>\n",
       "      <td>TCC</td>\n",
       "      <td>TCCC</td>\n",
       "      <td>70.0</td>\n",
       "      <td>0.909091</td>\n",
       "      <td>rph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430432</th>\n",
       "      <td>166a1092</td>\n",
       "      <td>3815879</td>\n",
       "      <td>P-13-2</td>\n",
       "      <td>TCC</td>\n",
       "      <td>TCCC</td>\n",
       "      <td>22.0</td>\n",
       "      <td>0.863636</td>\n",
       "      <td>rph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430448</th>\n",
       "      <td>166a1092</td>\n",
       "      <td>3815879</td>\n",
       "      <td>P-13-1</td>\n",
       "      <td>TCC</td>\n",
       "      <td>TCCC</td>\n",
       "      <td>21.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>rph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430449</th>\n",
       "      <td>166a1092</td>\n",
       "      <td>3815879</td>\n",
       "      <td>F-C6-2</td>\n",
       "      <td>TCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>86.0</td>\n",
       "      <td>0.965116</td>\n",
       "      <td>rph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430453</th>\n",
       "      <td>166a1092</td>\n",
       "      <td>3815879</td>\n",
       "      <td>C-E4-1</td>\n",
       "      <td>TCC</td>\n",
       "      <td>TCCC</td>\n",
       "      <td>36.0</td>\n",
       "      <td>0.914286</td>\n",
       "      <td>rph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430460</th>\n",
       "      <td>166a1092</td>\n",
       "      <td>3815879</td>\n",
       "      <td>W72</td>\n",
       "      <td>TCC</td>\n",
       "      <td>TCCC</td>\n",
       "      <td>33.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>rph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430463</th>\n",
       "      <td>166a1092</td>\n",
       "      <td>3815879</td>\n",
       "      <td>C-E4-2</td>\n",
       "      <td>TCC</td>\n",
       "      <td>TCCC</td>\n",
       "      <td>81.0</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>rph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430467</th>\n",
       "      <td>166a1092</td>\n",
       "      <td>3815879</td>\n",
       "      <td>P-8-2</td>\n",
       "      <td>TCC</td>\n",
       "      <td>TCCC</td>\n",
       "      <td>33.0</td>\n",
       "      <td>0.969697</td>\n",
       "      <td>rph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430470</th>\n",
       "      <td>166a1092</td>\n",
       "      <td>3815879</td>\n",
       "      <td>W62</td>\n",
       "      <td>TCC</td>\n",
       "      <td>TCCC</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>rph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430483</th>\n",
       "      <td>166a1092</td>\n",
       "      <td>3815879</td>\n",
       "      <td>F-C6-1</td>\n",
       "      <td>TCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>59.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>rph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430501</th>\n",
       "      <td>166a1092</td>\n",
       "      <td>3815879</td>\n",
       "      <td>P-8-1</td>\n",
       "      <td>TCC</td>\n",
       "      <td>TCCC</td>\n",
       "      <td>69.0</td>\n",
       "      <td>0.971014</td>\n",
       "      <td>rph</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             UID  position lineage  REF   ALT    DP        AF gene\n",
       "430400  166a1092   3815879  C-G4-1  TCC  TCCC  82.0  0.961538  rph\n",
       "430410  166a1092   3815879  C-G4-2  TCC  TCCC  70.0  0.909091  rph\n",
       "430432  166a1092   3815879  P-13-2  TCC  TCCC  22.0  0.863636  rph\n",
       "430448  166a1092   3815879  P-13-1  TCC  TCCC  21.0  1.000000  rph\n",
       "430449  166a1092   3815879  F-C6-2  TCC    TC  86.0  0.965116  rph\n",
       "430453  166a1092   3815879  C-E4-1  TCC  TCCC  36.0  0.914286  rph\n",
       "430460  166a1092   3815879     W72  TCC  TCCC  33.0  1.000000  rph\n",
       "430463  166a1092   3815879  C-E4-2  TCC  TCCC  81.0  0.950000  rph\n",
       "430467  166a1092   3815879   P-8-2  TCC  TCCC  33.0  0.969697  rph\n",
       "430470  166a1092   3815879     W62  TCC  TCCC  20.0  0.950000  rph\n",
       "430483  166a1092   3815879  F-C6-1  TCC    TC  59.0  1.000000  rph\n",
       "430501  166a1092   3815879   P-8-1  TCC  TCCC  69.0  0.971014  rph"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgdata[(mgdata.position==3815879)&~(mgdata.ALT.isnull())]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3815863    84\n",
       "3815800    48\n",
       "3815810    24\n",
       "3815879    12\n",
       "3815820     8\n",
       "3815813     6\n",
       "3815844     4\n",
       "3815916     2\n",
       "3815907     2\n",
       "3815823     2\n",
       "3815754     2\n",
       "3815943     2\n",
       "3815986     1\n",
       "3815949     1\n",
       "Name: position, dtype: int64"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgdata[(mgdata.position<3815987)&(mgdata.position>3815753)&~(mgdata.ALT.isnull())].position.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Check TAG to TAAs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Find every stop codon in E. coli genome (MG1655 version 3 - NC_000913.3), and categorize by Amber-(TAG) / Ochre-(TAA) / Opal-(TGA) (do not consider ncRNA, rRNA, tRNA, tmRNA, or pseudo-genes).  \n",
    "  \n",
    "Once this is done, compare the results to a list of TAG positions used in the C321 recoding effort, supplied originally by Gleb, but Aditya took this file and adjusted the positions so that they make sense in MG1655v3. It appears that they missed a codon when they recoded!  \n",
    "  \n",
    "Stop locations are recorded as the first position on the sense strand, regardless of if the gene is sense or antisense.  Strand is recorded as 1 for sense or -1 for antisense."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total stop codon counts in MG1655 gene annotations: \n",
      " TAA    2678\n",
      "TGA    1175\n",
      "TAG     287\n",
      "TTT       1\n",
      "Name: codon, dtype: int64\n",
      "\n",
      "Non-standard stop codons detected in genes:\n",
      "      codon  gene  position  strand\n",
      "1273   TTT  ralA   1413732       1\n",
      "\n",
      "TAGs matched from c321:\n",
      " TAG    286\n",
      "Name: codon, dtype: int64 \n",
      "\n",
      "TAGs missing from c321:\n",
      "      codon  gene  position  strand\n",
      "1293   TAG  tfaR   1432984       1 \n",
      "\n",
      "35 TAGs called in c321, but don't correspond to genes in latest annotation: [239378, 280024, 315244, 340089, 582152, 716388, 1286984, 1381789, 1425980, 1432273, 1529938, 1544384, 1651537, 1803567, 2012001, 2034536, 2169693, 2210944, 2453265, 2471105, 2784529, 2786729, 2787434, 2788238, 2994415, 2995314, 2997689, 3365664, 3583454, 3815863, 3862497, 4500043, 4507463, 4519014, 4569309]\n"
     ]
    }
   ],
   "source": [
    "# build a data-collection framework\n",
    "stopCodons = {'gene':[],'position':[],'strand':[],'codon':[]}\n",
    "nonProteinGenes = {}\n",
    "\n",
    "# read-in MG1655v3 sequence file\n",
    "for record in SeqIO.parse(\"NC_000913v3.fasta\",\"fasta\"):\n",
    "    MG1655 = record\n",
    "\n",
    "# read-in genbank file with gene records\n",
    "for record in SeqIO.parse(\"NC_000913v3.gb\",\"genbank\"): #Use Biopython to get genbank record info for each gene\n",
    "    \n",
    "    # record all non protein-coding genes\n",
    "    for f in record.features:\n",
    "        if f.type == \"ncRNA\":\n",
    "            nonProteinGenes[f.qualifiers[\"gene\"][0]] = \"ncRNA\"\n",
    "        if f.type == \"rRNA\":\n",
    "            nonProteinGenes[f.qualifiers[\"gene\"][0]] = \"rRNA\"\n",
    "        if f.type == \"tRNA\":\n",
    "            nonProteinGenes[f.qualifiers[\"gene\"][0]] = \"tRNA\"\n",
    "        if f.type == \"tmRNA\":\n",
    "            nonProteinGenes[f.qualifiers[\"gene\"][0]] = \"tmRNA\"\n",
    "    \n",
    "    # iterate through all genes again, eliminating pseudo and non-protein coding genes\n",
    "    for f in record.features:\n",
    "        if f.type == \"gene\":\n",
    "            gene = f.qualifiers[\"gene\"][0]\n",
    "            \n",
    "            # check for pseudo genes\n",
    "            pseudo = False\n",
    "            try:\n",
    "                if f.qualifiers[\"pseudo\"]:\n",
    "                    pseudo = True\n",
    "            except:\n",
    "                pass\n",
    "            \n",
    "            # if it is a real protein-coding gene, record stop codon and details\n",
    "            if gene not in nonProteinGenes.keys() and not pseudo:\n",
    "                strand = f.location.strand\n",
    "                if strand == -1:\n",
    "                    stop_loc = f.location.start + 1\n",
    "                    stop_seq = str(MG1655[stop_loc - 1 : stop_loc + 2].seq.reverse_complement())\n",
    "                elif strand == 1:\n",
    "                    stop_loc = f.location.end - 2\n",
    "                    stop_seq = str(MG1655[stop_loc - 1 : stop_loc + 2].seq)\n",
    "                else:\n",
    "                    print('ERROR: no sense detected')\n",
    "                stopCodons['gene'].append(gene)\n",
    "                stopCodons['position'].append(stop_loc)\n",
    "                stopCodons['strand'].append(strand)\n",
    "                stopCodons['codon'].append(stop_seq)\n",
    "\n",
    "# print out some statistics\n",
    "STOP = pd.DataFrame.from_dict(stopCodons)\n",
    "print(\"Total stop codon counts in MG1655 gene annotations: \\n\",STOP.codon.value_counts())\n",
    "print(\"\\nNon-standard stop codons detected in genes:\\n\",STOP[~STOP.codon.isin(['TAG','TGA','TAA'])])\n",
    "\n",
    "# read in stop codon positions from gleb via aditya and compare to TAGs found by my processing\n",
    "# of the Genbank file.\n",
    "c321_pos = pd.read_csv(\"mg1655v3_uag_locations.csv\")\n",
    "matched_TAG = STOP[(STOP.position.isin(c321_pos.coord))|(STOP.position.isin(c321_pos.coord-2))]\n",
    "unmatched_TAG = STOP[~(STOP.position.isin(matched_TAG.position))&(STOP.codon=='TAG')]\n",
    "\n",
    "print(\"\\nTAGs matched from c321:\\n\",matched_TAG.codon.value_counts(),'\\n\\nTAGs missing from c321:\\n',unmatched_TAG,'\\n')\n",
    "missing_TAG = list(x for x in c321_pos.coord if x not in list(matched_TAG.position) and x not in list(matched_TAG.position + 2))\n",
    "print(len(missing_TAG),'TAGs called in c321, but don\\'t correspond to genes in latest annotation:',missing_TAG)\n",
    "\n",
    "# Add TAGs that were noted in Aditya's file, but missing from my processing into the DataFrame \"STOPs\"\n",
    "for p in missing_TAG:\n",
    "    p = int(p)\n",
    "    if MG1655[p-1] == 'G':\n",
    "        stop_loc = p-2\n",
    "        strand = 1\n",
    "    elif MG1655[p-1] == 'C':\n",
    "        stop_loc = p\n",
    "        strand = -1\n",
    "    else:\n",
    "        print('ERROR: position ',p,' was found to be A or T')\n",
    "    STOP.loc[len(STOP)] = ['TAG','unannotated',stop_loc,strand]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Match called variants with stop codon list. There are two locations where two TAGs abut one another, from genes on opposite strands terminating next to each other or overlapping. This means that in two instances two TAGs are co-counted at positions:                                                      \n",
    "\n",
    "634747 - TAG on -1 strand (position of first nucleotide in codon)                                    \n",
    "634744 - TAG on +1 strand                    \"                                                      \n",
    "746723 - TAG on -1 strand                    \"                                                      \n",
    "746724 - TAG on +1 strand                    \"                                                      \n",
    "\n",
    "No mutations were found to any of these doubled up TAGs\n",
    "\n",
    "frame refers to the first, second, or third base of the codon (positive strand front to back, negative strand back to front)  \n",
    "  \n",
    "Note -- There is one mutation to ECNR2 at position 3269600 that happens to a TAG codon, so this mutation is contained in the cfp_TAG dataframe, but with no ALT entries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 321 TAG variant positions from c321, which times 84 lineages aligned, should be 26964 calls. Less  55 lines with no coverage leaves  26909 calls expected.\n",
      "Actual number of called variants is: 26875 , or 34 fewer than predicted.\n",
      "\n",
      "The positions UIDs hit fewer than  84  times are: \n",
      " [2012700  568004 1264970 3269602  315244 3365664]\n",
      "\n",
      "Underhit lineage positions:  34 \n",
      "       codon         gene  position  strand lineage REF  ALT     DP   AF  frame\n",
      "18459   TAG  unannotated    315244       1  F-F6-1   G  NaN   83.0  0.0      3\n",
      "2047    TAG        insF1    568004       1  C-F4-1   G  NaN    1.0  0.0      3\n",
      "5466    TAG         hemA   1264970       1  P-13-1   G  NaN   24.0  0.0      3\n",
      "5512    TAG         hemA   1264970       1   P-8-1   G  NaN   56.0  0.0      3\n",
      "5511    TAG         hemA   1264970       1   P-2-2   G  NaN   12.0  0.0      3\n",
      "5510    TAG         hemA   1264970       1   P-9-2   G  NaN   21.0  0.0      3\n",
      "5507    TAG         hemA   1264970       1   P-7-2   G  NaN   17.0  0.0      3\n",
      "5503    TAG         hemA   1264970       1   P-4-2   G  NaN   18.0  0.0      3\n",
      "5500    TAG         hemA   1264970       1   P-1-1   G  NaN   22.0  0.0      3\n",
      "5499    TAG         hemA   1264970       1   P-6-1   G  NaN   52.0  0.0      3\n",
      "5491    TAG         hemA   1264970       1  P-11-2   G  NaN   34.0  0.0      3\n",
      "5474    TAG         hemA   1264970       1   P-8-2   G  NaN   21.0  0.0      3\n",
      "5472    TAG         hemA   1264970       1   P-6-2   G  NaN   23.0  0.0      3\n",
      "5471    TAG         hemA   1264970       1   P-1-2   G  NaN   12.0  0.0      3\n",
      "5465    TAG         hemA   1264970       1   P-2-1   G  NaN   52.0  0.0      3\n",
      "5463    TAG         hemA   1264970       1  P-11-1   G  NaN  103.0  0.0      3\n",
      "5459    TAG         hemA   1264970       1   P-9-1   G  NaN   86.0  0.0      3\n",
      "5444    TAG         hemA   1264970       1   P-5-1   G  NaN   34.0  0.0      3\n",
      "5433    TAG         hemA   1264970       1  P-13-2   G  NaN   18.0  0.0      3\n",
      "5426    TAG         hemA   1264970       1   P-4-1   G  NaN   23.0  0.0      3\n",
      "5425    TAG         hemA   1264970       1   P-3-2   G  NaN   14.0  0.0      3\n",
      "5420    TAG         hemA   1264970       1  P-12-1   G  NaN   12.0  0.0      3\n",
      "5418    TAG         hemA   1264970       1  P-14-2   G  NaN   20.0  0.0      3\n",
      "5414    TAG         hemA   1264970       1  P-14-1   G  NaN   41.0  0.0      3\n",
      "5412    TAG         hemA   1264970       1   P-5-2   G  NaN   17.0  0.0      3\n",
      "5411    TAG         hemA   1264970       1  P-10-1   G  NaN   97.0  0.0      3\n",
      "5409    TAG         hemA   1264970       1  P-10-2   G  NaN   14.0  0.0      3\n",
      "5408    TAG         hemA   1264970       1   P-7-1   G  NaN   57.0  0.0      3\n",
      "5514    TAG         hemA   1264970       1   P-3-1   G  NaN   38.0  0.0      3\n",
      "5517    TAG         hemA   1264970       1  P-12-2   G  NaN   30.0  0.0      3\n",
      "5586    TAG         fliE   2012700      -1   P-3-2   C  NaN    1.0  0.0      1\n",
      "12855   TAG         yhaC   3269602       1  C-F4-1   G  NaN    1.0  0.0      3\n",
      "20298   TAG  unannotated   3365664       1   P-9-1   G  NaN   49.0  0.0      3\n",
      "20344   TAG  unannotated   3365664       1   P-9-2   G  NaN   15.0  0.0      3\n",
      "\n",
      "There are 2.0 ancestral mutations to non-TAG stop codons in C-F-P lines:\n",
      "         gene codon strand frame REF ALT count\n",
      "2370019  arnA   TGA      1     1   T   C    84\n",
      "3598557  livJ   TAA     -1     3   A   G    84 \n",
      "\n",
      "There are 11 new mutations to Ochre/Opal codons in C-F-P lines\n",
      "      codon  gene  position  strand lineage       REF      ALT     DP  \\\n",
      "66      TAA  yafE    232548       1   P-4-2         A        G   23.0   \n",
      "158     TAA  tesB    474302      -1  C-D4-1         T        A   65.0   \n",
      "8291    TAA  cheR   1968506      -1   P-3-2         A        C    2.0   \n",
      "9406    TAA  yeiB   2241812      -1  C-C4-1         A        G   53.0   \n",
      "87      TAA  ypjK   2770672       1  P-14-1  GAAAAAAT  GAAAAAT   21.0   \n",
      "10265   TAA  gshB   3092826       1  C-G4-2         T        C  130.0   \n",
      "10318   TAA  gshB   3092826       1  C-E4-1         T        C   69.0   \n",
      "11504   TAA   fis   3411565       1  C-B3-2         T        C   77.0   \n",
      "11518   TAA   fis   3411565       1  C-B3-1         T        C  116.0   \n",
      "11538   TAA   fis   3411565       1  C-D3-2         T        C   39.0   \n",
      "11542   TAA   fis   3411565       1  C-D3-1         T        C   60.0   \n",
      "\n",
      "             AF  frame  \n",
      "66     1.000000      2  \n",
      "158    1.000000      2  \n",
      "8291   1.000000      3  \n",
      "9406   1.000000      3  \n",
      "87     1.000000     -6  \n",
      "10265  1.000000      1  \n",
      "10318  0.971014      1  \n",
      "11504  1.000000      1  \n",
      "11518  1.000000      1  \n",
      "11538  1.000000      1  \n",
      "11542  1.000000      1   \n",
      "\n",
      "There are 7 new mutations to stop codons in ECNR2 lines\n",
      "      codon   gene  position  strand lineage       REF        ALT     DP  \\\n",
      "4573    TAA    mlc   1667344      -1  E-F1-1         T          C   41.0   \n",
      "4611    TAA   pykF   1757108       1  E-A2-2         T          C   69.0   \n",
      "4693    TAA   pykF   1757108       1  E-A2-1         T          C   26.0   \n",
      "91      TAA  insB1   1978516      -1  E-H2-1       ACT         AT    2.0   \n",
      "85      TAA    mug   3214962      -1  E-H2-1  GTTTTTTA  GTTTTTTTA   23.0   \n",
      "10929   TAG   yhaC   3269600       1  E-G1-1         T          C   39.0   \n",
      "11672   TAA   acrF   3418135       1  E-H2-2         T          C  119.0   \n",
      "\n",
      "             AF  frame  \n",
      "4573   1.000000      1  \n",
      "4611   1.000000      1  \n",
      "4693   0.961538      1  \n",
      "91     1.000000     -1  \n",
      "85     1.000000     -4  \n",
      "10929  1.000000      1  \n",
      "11672  1.000000      1   \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# create dataframe to hold mutations at STOP positions\n",
    "mg_STOP = pd.DataFrame()\n",
    "\n",
    "# find and record all STOP codon variant info.\n",
    "# record any length variant call for variants called at stop positions\n",
    "for i in range(0,3):\n",
    "    df = STOP.copy()\n",
    "    df['position'] = df['position'].apply(lambda x: x + i)\n",
    "    dg = mgdata[mgdata.position.isin(df.position)].loc[:,['position','lineage','REF','ALT','DP','AF']]\n",
    "    df = pd.merge(df,dg,left_on='position',right_on='position')\n",
    "    df['frame'] = i+1\n",
    "    mg_STOP = mg_STOP.append(df)\n",
    "    \n",
    "# record long variant calls up to 10nt upstream the STOP codon if the variant call overlaps the stop codon\n",
    "for i in range(-10,0):\n",
    "    df = STOP.copy()\n",
    "    df['position'] = df['position'].apply(lambda x: x + i)\n",
    "    dg = mgdata[mgdata.position.isin(df.position)&(mgdata.REF.str.len() > -i)]\n",
    "    dg = dg.loc[:,['position','lineage','REF','ALT','DP','AF']]\n",
    "    df = pd.merge(df,dg,left_on='position',right_on='position')\n",
    "    df['frame'] = i+1\n",
    "    mg_STOP = mg_STOP.append(df)\n",
    "\n",
    "# break STOPs dataframe into sub-sets for CFP and E populations and CFP into TAGs vs TAATGAs\n",
    "cfp_TAG = mg_STOP[mg_STOP.lineage.str[0].isin(['C','F','P'])&(mg_STOP.codon=='TAG')]\n",
    "cfp_TAATGA = mg_STOP[(mg_STOP.lineage.str[0].isin(['C','F','P']))&(mg_STOP.codon.isin(['TAA','TGA']))]\n",
    "e_STOP = mg_STOP[mg_STOP.lineage.str[0]=='E']\n",
    "\n",
    "# Find reversions to TAGs in CFP populatinos by counting diversion from expected muations\n",
    "# Note all TAG reversions that don't have zero read depth\n",
    "underhit_TAG = pd.DataFrame()\n",
    "for p in cfp_TAG.groupby(['position','ALT']).filter(lambda x: len(x) < cfp_count).position.unique():\n",
    "    # clean out positions found just because of poor read depth\n",
    "    hits = cfp_TAG[(cfp_TAG.position==p)&(cfp_TAG.ALT.notnull())].shape[0]\n",
    "    no_read = cfp_TAG[(cfp_TAG.position==p)&(cfp_TAG.DP.isnull())].shape[0]\n",
    "    if hits + no_read != cfp_count:\n",
    "        underhit_TAG = pd.concat([underhit_TAG,cfp_TAG.groupby(['position']).get_group(p)])\n",
    "underhit_TAG_real = underhit_TAG[(underhit_TAG.ALT.isnull())&(underhit_TAG.DP.notnull())].sort_values(by='position')\n",
    "\n",
    "# Analyze TGA and TAA mutations among evolving CFP populations\n",
    "# Divide cfp TGA and TAA mutations into two classes: ancestral (from c321) and new (appeared during evolution)\n",
    "cfp_TAATGAmut = cfp_TAATGA[(cfp_TAATGA.DP.notnull())&(cfp_TAATGA.ALT.notnull())&(cfp_TAATGA.AF>fraction)]\n",
    "cfp_TAATGAancestral = cfp_TAATGAmut.groupby(['position','ALT']).filter(lambda x: len(x) >= cfp_count)\n",
    "cfp_TAATGAnew = cfp_TAATGAmut.groupby(['position','ALT']).filter(lambda x: len(x) < cfp_count)\n",
    "ancestral = pd.DataFrame(columns=['gene','codon','strand','frame','REF','ALT','count'])\n",
    "for p in cfp_TAATGAancestral.position.unique():\n",
    "    ancestral.loc[p,'gene']=cfp_TAATGAancestral[cfp_TAATGAancestral.position==p].gene.unique()[0]\n",
    "    ancestral.loc[p,'codon']=cfp_TAATGAancestral[cfp_TAATGAancestral.position==p].codon.unique()[0]\n",
    "    ancestral.loc[p,'strand']=cfp_TAATGAancestral[cfp_TAATGAancestral.position==p].strand.unique()[0]\n",
    "    ancestral.loc[p,'frame']=cfp_TAATGAancestral[cfp_TAATGAancestral.position==p].frame.unique()[0]\n",
    "    ancestral.loc[p,'REF']=cfp_TAATGAancestral[cfp_TAATGAancestral.position==p].REF.unique()[0]\n",
    "    ancestral.loc[p,'ALT']=cfp_TAATGAancestral[cfp_TAATGAancestral.position==p].ALT.unique()[0]\n",
    "    ancestral.loc[p,'count']=cfp_TAATGAancestral[cfp_TAATGAancestral.position==p].ALT.count()\n",
    "\n",
    "cfp_TAATGAnew = cfp_TAATGAnew.sort_values(by='position')\n",
    "\n",
    "# find all stop mutations in ECNR2 lines\n",
    "e_STOPmut = e_STOP[(e_STOP.ALT.notnull())&(e_STOP.AF>fraction)&(e_STOP.DP.notnull())].sort_values(by='position')\n",
    "\n",
    "# print out some statistics\n",
    "p = len(list(matched_TAG.position.unique())+missing_TAG)\n",
    "s = cfp_TAG.lineage.nunique()\n",
    "v = cfp_TAG.ALT.count()\n",
    "# NOTE -- there are two positions with no coverage from position 3269600 that is hit in ECNR2 that need to be \n",
    "# subtracted from the count of no-coverage alleles for the numbers to add up.\n",
    "nc = cfp_TAG[cfp_TAG.DP.isnull()].shape[0]-2\n",
    "print('There are',p,'TAG variant positions from c321, which times',s,'lineages aligned, should be',p*s\n",
    "      ,'calls. Less ',nc,'lines with no coverage leaves ',p*s-nc,'calls expected.')\n",
    "print('Actual number of called variants is:',v,', or',p*s-nc-v,'fewer than predicted.\\n')\n",
    "print('The positions UIDs hit fewer than ',s,' times are: \\n',underhit_TAG.position.unique())\n",
    "print('\\nUnderhit lineage positions: ',underhit_TAG_real.position.count(),'\\n',underhit_TAG_real)\n",
    "print('\\nThere are',len(cfp_TAATGAancestral)/cfp_count,'ancestral mutations to non-TAG stop codons in C-F-P lines:')\n",
    "print(ancestral,'\\n')\n",
    "print('There are',len(cfp_TAATGAnew.position),'new mutations to Ochre/Opal codons in C-F-P lines')\n",
    "print(cfp_TAATGAnew,'\\n')\n",
    "print('There are',len(e_STOPmut.position),'new mutations to stop codons in ECNR2 lines')\n",
    "print(e_STOPmut,'\\n')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Scan for frameshift mutations upstream of the end of the gene\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/timothywannier/anaconda/lib/python3.6/site-packages/Bio/Seq.py:2095: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  BiopythonWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The following positions are ancestral frameshifts in CFP populations that maintained complete penetrance after evolution:\n",
      "         v2gene       v3gene        REF        ALT offset codon codon_alt  \\\n",
      "183606    cdaR         cdaR        GAT         GT      1   TAG       TAG   \n",
      "675600    ybeQ         ybeQ        CTG         CG      1   TAG       TGA   \n",
      "1006412   zapC         zapC        TGG         TG      1   TAA       TGA   \n",
      "1757751   ldtE         ldtE    CTCTGGT         CT      5   TAG       TAG   \n",
      "2824495   mltB         mltB        TGT         TT      1   TAG       TAA   \n",
      "3134810   yghT         yghT        GTG         GG      1   TAG       TAA   \n",
      "3330353   dacB         dacB        GTT       GTTT     -1   TAG       TGA   \n",
      "3862523   glvC  unannotated      TGCCC         TC      3   TAG       TAG   \n",
      "4424542   yjfY         yjfY        GCA         GA      1   TAG       TAA   \n",
      "4447866   tamB         tamB        ATT         AT      1   TAG       TAA   \n",
      "4619469   deoA         deoA  LONG:11bp         CG      9   TAA       TAA   \n",
      "1655355   ynfA         ynfA    GCCCCCA     GCCCCA      1   TAG       TAA   \n",
      "3970026   wzzE         wzzE   CGGGGGGC  CGGGGGGGC     -1   TAG       TGA   \n",
      "1208533   tfaP         tfaE        TCA         TA      1   TAG       TAA   \n",
      "3862515   glvC  unannotated        GCC         GC      1   TAG       TGA   \n",
      "\n",
      "        nt_upstream new_downstream strand population  count  \n",
      "183606           12             67      1        CFP   84.0  \n",
      "675600           28            134     -1        CFP   84.0  \n",
      "1006412          80             19      1        CFP   84.0  \n",
      "1757751          28            108     -1        CFP   84.0  \n",
      "2824495           2             68     -1        CFP   84.0  \n",
      "3134810          11             61      1        CFP   84.0  \n",
      "3330353          41              4      1        CFP   84.0  \n",
      "3862523          24             22     -1          F   29.0  \n",
      "4424542          24            167     -1        CFP   84.0  \n",
      "4447866          23            118      1        CFP   84.0  \n",
      "4619469          80             70      1        CFP   84.0  \n",
      "1655355           6             29     -1          F   29.0  \n",
      "3970026          49             67      1          F   30.0  \n",
      "1208533          14            293     -1        CFP  166.0  \n",
      "3862515          16              8     -1        CFP   84.0   \n",
      "\n",
      "The following positions are ancestral frameshifts in CFP populations that showed reversion mutations after evolution:\n",
      "           v2gene       v3gene       REF                    ALT codon  \\\n",
      "3970026-1   wzzE         wzzE  CGGGGGGC              CGGGGGGGC   TAG   \n",
      "3970026-2   wzzE         wzzE  CGGGGGGC  LONG:13,HASH:18bcc667   TAG   \n",
      "3970026-3   wzzE         wzzE  CGGGGGGC             CGGGGGGGGC   TAG   \n",
      "3970026-4   wzzE         wzzE  CGGGGGGC  LONG:12,HASH:f3605d4d   TAG   \n",
      "3970026-5   wzzE         wzzE  CGGGGGGC  LONG:11,HASH:474afe0d   TAG   \n",
      "3970026-6   wzzE         wzzE  CGGGGGGC                    NaN   TAG   \n",
      "1423525-1   rzpR         rzoR  TCCCCCCG              TCCCCCCCG   TAA   \n",
      "1423525-2   rzpR         rzoR  TCCCCCCG               CCCCCCCG   TAA   \n",
      "3862523-1   glvC  unannotated     TGCCC                     TC   TAG   \n",
      "3862523-2   glvC  unannotated     TGCCC                    NaN   TAG   \n",
      "\n",
      "          codon_alt nt_upstream new_downstream strand population  count  \\\n",
      "3970026-1       TGA          49             67      1        CFP   76.0   \n",
      "3970026-2       TGA          49             13      1        CFP    2.0   \n",
      "3970026-3       TGA          49             10      1        CFP    6.0   \n",
      "3970026-4       TGA          49             70      1        CFP    2.0   \n",
      "3970026-5       TAG          49             52      1        CFP    1.0   \n",
      "3970026-6       TAG          49             49      1        CFP    1.0   \n",
      "1423525-1       TAA          58            118      1          P   26.0   \n",
      "1423525-2       TAA          58             58      1          P    4.0   \n",
      "3862523-1       TAG          24             22     -1        CFP   83.0   \n",
      "3862523-2       TAG          24             22     -1        CFP    1.0   \n",
      "\n",
      "           offset  \n",
      "3970026-1    -1.0  \n",
      "3970026-2    -5.0  \n",
      "3970026-3    -2.0  \n",
      "3970026-4    -4.0  \n",
      "3970026-5    -3.0  \n",
      "3970026-6     0.0  \n",
      "1423525-1    -1.0  \n",
      "1423525-2     0.0  \n",
      "3862523-1     3.0  \n",
      "3862523-2     0.0   \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# record variant calls up to 90nt upstream (but not including) the STOP codon if the variant call indicates \n",
    "# a frameshift\n",
    "buffer = 1000\n",
    "window = 100\n",
    "\n",
    "mg_FRAMESHIFT = pd.DataFrame()\n",
    "for i in range(-window,-1):\n",
    "    df = STOP.copy()\n",
    "    # adjust the STOP position\n",
    "    df['adjustment'] = df['strand'].apply(lambda x: x * i + 1)  \n",
    "    df['position'] = df['position'] + df['adjustment']\n",
    "    # pull frameshift mutations\n",
    "    dg = mvc[mvc.position.isin(df.position) & mvc.ALT.notnull() & (mvc.REF.str.len() != mvc.ALT.str.len())]\n",
    "    dg = dg.loc[:,['position','lineage','REF','ALT','DP','AF']]\n",
    "    df = pd.merge(df,dg,left_on='position',right_on='position')\n",
    "    mg_FRAMESHIFT = mg_FRAMESHIFT.append(df)\n",
    "\n",
    "# add in a couple of ancestral variants:\n",
    "mg_FRAMESHIFT.loc[mg_FRAMESHIFT.shape[0]] = ['TAA','rzoR',1423525,1,-58,'P-G0','TCCCCCCG','TCCCCCCCG',50,1]\n",
    "mg_FRAMESHIFT.loc[mg_FRAMESHIFT.shape[0]] = ['TAG','unannotated',3862515,1,-18,'C-G0','GCC','GC',8,1]\n",
    "\n",
    "lineages = ['E','C','F','P','CFP']\n",
    "l_samples = {'E':e_samples,'C':c_samples,'F':f_samples,'P':p_samples,'CFP':cfp_samples}\n",
    "l_count = {'E':e_count,'C':c_count,'F':f_count,'P':p_count,'CFP':cfp_count}\n",
    "l_ancestors = {'E':'W72','C':'C-G0','F':'F-G0','P':'P-G0','CFP':'C-G0'}\n",
    "\n",
    "F_ancestral = pd.DataFrame(columns=['v2gene','v3gene','REF','ALT','offset','codon','codon_alt',\n",
    "                                    'nt_upstream','new_downstream','strand','population'])\n",
    "F_reversion = pd.DataFrame(columns=['v2gene','v3gene','REF','ALT','codon','codon_alt','nt_upstream',\n",
    "                           'new_downstream','strand','population'])\n",
    "\n",
    "# Populate ancestral DF with info on stop codon and alternate for populations with no reversions\n",
    "for line in lineages:\n",
    "    FS = mg_FRAMESHIFT[mg_FRAMESHIFT.lineage.isin(l_samples[line])]\n",
    "    A_pos = list(mg_FRAMESHIFT[mg_FRAMESHIFT.lineage == l_ancestors[line]].position.unique())\n",
    "    A_FS = FS[FS.position.isin(A_pos)]\n",
    "    mg_FS = mgdata[(mgdata.position.isin(A_pos)) & mgdata.lineage.isin(l_samples[line])]\n",
    "    A_FS = pd.merge(mg_FS,A_FS,how='outer', left_on=['lineage','position','REF','ALT','DP','AF'],\n",
    "                                            right_on=['lineage','position','REF','ALT','DP','AF'])\n",
    "    A_FS['position'] = A_FS['position'].astype(int) # prevents floats from messing up processing\n",
    "    A_FS_norev = A_FS.groupby(['position','ALT']).filter(lambda x: len(x) == l_count[line])\n",
    "    if line == 'CFP':\n",
    "        A_FS_norev = A_FS.groupby(['position','ALT']).filter(lambda x: len(x) >= l_count[line])\n",
    "    A_FS_rev = A_FS.groupby(['position','ALT']).filter(lambda x: len(x) < l_count[line])\n",
    "    \n",
    "    for p in A_FS_norev.position.unique():\n",
    "        F_ancestral.loc[p,'v2gene']=A_FS[A_FS.position==p].gene_x.dropna().unique()[0]\n",
    "        F_ancestral.loc[p,'v3gene']=A_FS[A_FS.position==p].gene_y.dropna().unique()[0]\n",
    "        F_ancestral.loc[p,'codon']=A_FS[A_FS.position==p].codon.dropna().unique()[0]\n",
    "        # get stop codon that the frameshift switches to:\n",
    "        F_ancestral.loc[p,'REF'] = r = A_FS[A_FS.position==p].REF.unique()[0]\n",
    "        F_ancestral.loc[p,'ALT'] = a = A_FS[A_FS.position==p].ALT.unique()[0]\n",
    "        if 'LONG' in r:\n",
    "            r = 'X' * int(re.findall('\\d+', r)[0])\n",
    "        if 'LONG' in a:\n",
    "            a = 'X' * int(re.findall('\\d+', a)[0])\n",
    "        F_ancestral.loc[p,'strand']=s=A_FS[A_FS.position==p].strand.dropna().unique()[0]\n",
    "        u = abs(A_FS[A_FS.position==p].adjustment.dropna().unique()[0])\n",
    "        F_ancestral.loc[p,'offset']=o=(len(r) - len(a))\n",
    "        q = int(p - 1 + ((u % 3) + o) * s)\n",
    "        if s == 1:\n",
    "            n = MG1655[q : q + buffer]\n",
    "            aa = n.seq.translate()\n",
    "            l = len(aa.split('*')[0]) * 3\n",
    "            F_ancestral.loc[p,'codon_alt'] = str(MG1655[q + l : q + l + 3].seq)\n",
    "        else:\n",
    "            u -= 2 # resize u for accurate reporting of upstream bases on reverse strand\n",
    "            n = MG1655[q - buffer : q]\n",
    "            aa = n.seq.reverse_complement().translate()\n",
    "            l = len(aa.split('*')[0]) * 3\n",
    "            F_ancestral.loc[p,'codon_alt'] = str(MG1655[q - l - 3 : q - l].reverse_complement().seq)\n",
    "            l += o # resize l for accurate reporting of new downstream bases on reverse strand\n",
    "        F_ancestral.loc[p,'nt_upstream'] = u\n",
    "        F_ancestral.loc[p,'new_downstream'] = l + 1\n",
    "        F_ancestral.loc[p,'population'] = line\n",
    "        F_ancestral.loc[p,'count'] = A_FS[A_FS.position==p].ALT.count()\n",
    "\n",
    "# Populate ancestral DF with info on stop codon and alternate for populations with reversions\n",
    "    for p in A_FS_rev.position.unique():\n",
    "        i = 0\n",
    "        alts = A_FS[A_FS.position==p].ALT.unique()\n",
    "        for alt in alts:\n",
    "            i += 1\n",
    "            pos = str(p) + '-' + str(i)\n",
    "            F_reversion.loc[pos,'v2gene']=A_FS[A_FS.position==p].gene_x.dropna().unique()[0]\n",
    "            F_reversion.loc[pos,'v3gene']=A_FS[A_FS.position==p].gene_y.dropna().unique()[0]\n",
    "            F_reversion.loc[pos,'codon']=A_FS[A_FS.position==p].codon.dropna().unique()[0]\n",
    "            # get stop codon that the frameshift switches to:\n",
    "            F_reversion.loc[pos,'REF'] = r = A_FS[A_FS.position==p].REF.unique()[0]\n",
    "            F_reversion.loc[pos,'ALT'] = a = alt\n",
    "            if 'LONG' in r:\n",
    "                r = 'X' * int(re.findall('\\d+', r)[0])\n",
    "            if isinstance(a,float):\n",
    "                a = r\n",
    "                F_reversion.loc[pos,'count'] = A_FS[A_FS.position==p].ALT.isnull().sum()\n",
    "            else:\n",
    "                if 'LONG' in a:\n",
    "                    a = 'X' * int(re.findall('\\d+', a)[0])\n",
    "                F_reversion.loc[pos,'count'] = A_FS[(A_FS.position==p) & (A_FS.ALT==alt)].shape[0]\n",
    "            F_reversion.loc[pos,'strand']=s=A_FS[A_FS.position==p].strand.dropna().unique()[0]\n",
    "            u = abs(A_FS[A_FS.position==p].adjustment.dropna().unique()[0])\n",
    "            F_reversion.loc[pos,'offset']=o=(len(r) - len(a))\n",
    "            q = int(p - 1 + ((u % 3) + o) * s)\n",
    "            if s == 1:\n",
    "                n = MG1655[q : q + buffer]\n",
    "                aa = n.seq.translate()\n",
    "                l = len(aa.split('*')[0]) * 3\n",
    "                F_reversion.loc[pos,'codon_alt'] = str(MG1655[q + l : q + l + 3].seq)\n",
    "            else:\n",
    "                u -= 2 # resize u for accurate reporting of upstream bases on reverse strand\n",
    "                n = MG1655[q - buffer : q]\n",
    "                aa = n.seq.reverse_complement().translate()\n",
    "                l = len(aa.split('*')[0]) * 3\n",
    "                F_reversion.loc[pos,'codon_alt'] = str(MG1655[q - l - 3 : q - l].reverse_complement().seq)\n",
    "                l += o # resize l for accurate reporting of new downstream bases on reverse strand\n",
    "            F_reversion.loc[pos,'nt_upstream'] = u\n",
    "            F_reversion.loc[pos,'new_downstream'] = l + 1\n",
    "            F_reversion.loc[pos,'population'] = line\n",
    "\n",
    "print('The following positions are ancestral frameshifts in CFP populations that maintained complete penetrance',\n",
    "      'after evolution:\\n',F_ancestral,'\\n')\n",
    "print('The following positions are ancestral frameshifts in CFP populations that showed reversion mutations',\n",
    "      'after evolution:\\n',F_reversion,'\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/timothywannier/anaconda/lib/python3.6/site-packages/Bio/Seq.py:2095: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  BiopythonWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "new frameshift mutations that impact stop codon usage:\n",
      "                                                     lineages v2gene  \\\n",
      "115797-1                                            [P-10-1]   hofB   \n",
      "578096-1                                            [E-E2-1]   rrrD   \n",
      "635369-1                            [P-10-2, C-A4-1, C-A4-2]   ybdM   \n",
      "821493-1                                            [E-B2-1]   ybhM   \n",
      "1084580-1                                           [C-D3-1]   efeB   \n",
      "1189846-1                                   [C-C4-2, F-B6-1]   phoP   \n",
      "1199759-1                                   [F-G5-2, F-G5-1]   intE   \n",
      "1417538-1                                           [E-G1-1]   racC   \n",
      "1639535-1                                           [C-F3-2]   rrrQ   \n",
      "1660484-1                                           [P-14-2]   ynfE   \n",
      "1931971-1                   [F-D5-1, F-D5-2, F-G5-1, F-G5-2]   purT   \n",
      "1967130-1                                           [E-A2-1]   cheY   \n",
      "2024571-1                                           [F-F6-2]   rcsA   \n",
      "2186661-1                                           [F-C5-1]   rcnA   \n",
      "2297107-1                                           [F-E6-1]   ccmA   \n",
      "2367984-1                             [P-3-2, P-8-2, E-F1-2]   arnC   \n",
      "2770672-1                                           [P-14-1]   ypjK   \n",
      "2984447-1                                           [E-E1-1]   yqeF   \n",
      "3003473-1                                   [C-B3-1, P-12-2]   xdhB   \n",
      "3019150-1                                           [C-D4-2]   ygfK   \n",
      "3411546-1                   [C-G3-2, C-F3-1, C-G3-1, C-F3-2]    fis   \n",
      "3799382-1                     [F-G6-2, P-1-2, F-G6-1, P-6-1]   waaZ   \n",
      "3815879-1  [C-G4-1, C-G4-2, P-13-2, P-13-1, C-E4-1, W72, ...    rph   \n",
      "3815879-2                                   [F-C6-2, F-C6-1]    rph   \n",
      "3815907-1                                   [F-E6-1, F-E6-2]    rph   \n",
      "3815916-1                                   [F-F5-2, F-F5-1]    rph   \n",
      "3815943-1                                   [F-G5-1, F-G5-2]    rph   \n",
      "3867032-1                                           [C-F4-2]   ibpA   \n",
      "3902308-1                                   [E-B2-2, F-B6-2]   bglB   \n",
      "3938217-1                   [C-C3-2, C-C3-1, C-E3-1, C-E3-2]   rbsK   \n",
      "3950463-1                                   [E-C1-2, E-C1-1]   ilvX   \n",
      "3950463-2                                   [E-F1-2, E-E1-2]   ilvX   \n",
      "3980619-1                                           [P-14-2]   wecG   \n",
      "4293099-1                                    [E-E1-2, P-5-2]   nrfE   \n",
      "4415881-1                     [P-3-1, E-A1-1, E-A1-2, P-3-2]   aidB   \n",
      "4472418-1                     [P-10-2, C-D4-2, P-7-2, P-4-2]   pyrL   \n",
      "4488572-1                           [P-12-2, E-F1-1, F-C6-1]   yjgR   \n",
      "\n",
      "                v3gene         REF         ALT offset codon codon_alt  \\\n",
      "115797-1          hofB     CAAAAAG      CAAAAG      1   TAA       TAG   \n",
      "578096-1          rrrD     TGGGGGC    TGGGGGGC     -1   TGA       TAG   \n",
      "635369-1          ybdN     TCCCCCA    TCCCCCCA     -1   TAA       TAG   \n",
      "821493-1          ybhM   GTTTTTTTC  GTTTTTTTTC     -1   TAG       TAA   \n",
      "1084580-1         efeB   ATTTTTTTG  ATTTTTTTTG     -1   TAA       TAA   \n",
      "1189846-1         phoP   ATTTTTTTG  ATTTTTTTTG     -1   TGA       ATC   \n",
      "1199759-1         intE     GTTTTTG    GTTTTTTG     -1   TGA       TAA   \n",
      "1417538-1         racC  CTTTTTTTTC   CTTTTTTTC      1   TGA       TAA   \n",
      "1639535-1         rrrQ     TCCCCCA    TCCCCCCA     -1   TGA       TAG   \n",
      "1660484-1         ynfE    GAAAAAAT   GAAAAAAAT     -1   TAA       TAA   \n",
      "1931971-1         purT     TGGGGGT    TGGGGGGT     -1   TAA       TGA   \n",
      "1967130-1         cheY     GCCCCCG    GCCCCCCG     -1   TGA       TGA   \n",
      "2024571-1         rcsA     ATTTTTG      ATTTTG      1   TAA       TGA   \n",
      "2186661-1         rcnA     GCCCCCT    GCCCCCCT     -1   TAA       TAA   \n",
      "2297107-1         ccmA   TCCCCCCCT  TCCCCCCCCT     -1   TGA       TGA   \n",
      "2367984-1         arnC     GCCCCCG    GCCCCCCG     -1   TAA       TGA   \n",
      "2770672-1         ypjK    GAAAAAAT     GAAAAAT      1   TAA       TAA   \n",
      "2984447-1         yqeF     GCCCCCG      GCCCCG      1   TAG       TGA   \n",
      "3003473-1         xdhB    CGGGGGGA   CGGGGGGGA     -1   TGA       TAA   \n",
      "3019150-1         ygfK         GAG        GAAG     -1   TAA       CGC   \n",
      "3411546-1          fis    GAAAAAAT     GAAAAAT      1   TAA       TGA   \n",
      "3799382-1         waaZ  ATTTTTTTTC   ATTTTTTTC      1   TAA       TAA   \n",
      "3815879-1  unannotated         TCC        TCCC     -1   TAG       TGA   \n",
      "3815879-2  unannotated         TCC          TC      1   TAG       TAG   \n",
      "3815907-1  unannotated         GTA          GA      1   TAG       TAG   \n",
      "3815916-1  unannotated         CAT          CT      1   TAG       TAG   \n",
      "3815943-1  unannotated         TCC          TC      1   TAG       TAG   \n",
      "3867032-1         ibpA    GTTTTTTC   GTTTTTTTC     -1   TAA       TAA   \n",
      "3902308-1         bglB  ATTTTTTTTA   ATTTTTTTA      1   TAA       TAA   \n",
      "3938217-1         rbsK         GAG        GAAG     -1   TGA       TAA   \n",
      "3950463-1         ilvX     CGGGGGT      CGGGGT      1   TAA       TAA   \n",
      "3950463-2         ilvX     CGGGGGT    CGGGGGTT     -1   TAA       TGA   \n",
      "3980619-1         wecG         CTT          CT      1   TGA       TAA   \n",
      "4293099-1         nrfE   TGGGGGGGA  TGGGGGGGGA     -1   TAA       TGA   \n",
      "4415881-1         aidB    CGGGGGGA   CGGGGGGGA     -1   TAA       TGA   \n",
      "4472418-1         pyrL  CAAAAAAAAG   CAAAAAAAG      1   TAA       TAA   \n",
      "4488572-1         yjgR   TCCCCCCCA  TCCCCCCCCA     -1   TAA       TGA   \n",
      "\n",
      "          nt_upstream new_downstream strand count  \n",
      "115797-1           81             17     -1     1  \n",
      "578096-1           12            472      1     1  \n",
      "635369-1           18            618     -1     3  \n",
      "821493-1           11             43      1     1  \n",
      "1084580-1          65             82      1     1  \n",
      "1189846-1          68            999     -1     2  \n",
      "1199759-1          78              9     -1     2  \n",
      "1417538-1          48             17     -1     1  \n",
      "1639535-1           9            501     -1     1  \n",
      "1660484-1           9             61      1     1  \n",
      "1931971-1          86             34      1     4  \n",
      "1967130-1          80             63     -1     1  \n",
      "2024571-1          18             28      1     1  \n",
      "2186661-1          78            637      1     1  \n",
      "2297107-1          84             36     -1     1  \n",
      "2367984-1          53             49      1     3  \n",
      "2770672-1           7             16      1     1  \n",
      "2984447-1          34             14     -1     1  \n",
      "3003473-1          17            493      1     2  \n",
      "3019150-1           6           1000      1     1  \n",
      "3411546-1          19             13      1     4  \n",
      "3799382-1          35            125     -1     4  \n",
      "3815879-1          14             42     -1    10  \n",
      "3815879-2          14             83     -1     2  \n",
      "3815907-1          42            110     -1     2  \n",
      "3815916-1          51            119     -1     2  \n",
      "3815943-1          78            146     -1     2  \n",
      "3867032-1          21             27     -1     1  \n",
      "3902308-1          17             56     -1     2  \n",
      "3938217-1           4             79      1     4  \n",
      "3950463-1          92             25      1     2  \n",
      "3950463-2          92             13      1     2  \n",
      "3980619-1          75            157      1     1  \n",
      "4293099-1          69             67      1     2  \n",
      "4415881-1          17             31      1     4  \n",
      "4472418-1          17             59     -1     4  \n",
      "4488572-1           9             33     -1     3  \n"
     ]
    }
   ],
   "source": [
    "# analyze new frameshift mutations that appeared during the evolutionary process\n",
    "ancestral_positions = list(set([int(x[:-2]) for x in F_reversion.index.tolist()]))\n",
    "ancestral_positions += F_ancestral.index.tolist()\n",
    "all_F_new = mg_FRAMESHIFT[~mg_FRAMESHIFT.position.isin(ancestral_positions)]\n",
    "all_F_new = pd.merge(mgdata[mgdata.position.isin(list(all_F_new.position.unique()))],\n",
    "                     all_F_new,how='outer',left_on=['lineage','position','REF','ALT','DP','AF'],\n",
    "                                           right_on=['lineage','position','REF','ALT','DP','AF'])\n",
    "\n",
    "F_new = pd.DataFrame(columns=['lineages','v2gene','v3gene','REF','ALT','offset','codon','codon_alt','nt_upstream',\n",
    "                              'new_downstream','strand','count'])\n",
    "F_new.astype(object)\n",
    "for p in all_F_new.position.unique():\n",
    "    i = 0\n",
    "    alts = all_F_new[all_F_new.position==p].ALT.dropna().unique()\n",
    "    for alt in alts:\n",
    "        i += 1\n",
    "        pos = str(p) + '-' + str(i)\n",
    "        F_new.set_value(pos,'lineages',list(all_F_new[(all_F_new.position == p) & \n",
    "                                                   (all_F_new.ALT == alt)].lineage.unique()))\n",
    "        F_new.set_value(pos,'v2gene',all_F_new[all_F_new.position == p].gene_x.dropna().unique()[0])\n",
    "        F_new.set_value(pos,'v3gene',all_F_new[all_F_new.position == p].gene_y.dropna().unique()[0])\n",
    "        F_new.set_value(pos,'codon',all_F_new[all_F_new.position == p].codon.dropna().unique()[0])\n",
    "        # get stop codon that the frameshift switches to:\n",
    "        r = all_F_new[all_F_new.position == p].REF.unique()[0]\n",
    "        F_new.set_value(pos,'REF',r)\n",
    "        F_new.set_value(pos,'ALT',alt)\n",
    "        if 'LONG' in r:\n",
    "            r = 'X' * int(re.findall('\\d+', r)[0])\n",
    "        if isinstance(alt,float):\n",
    "            alt = r\n",
    "            F_new.set_value(pos,'count',all_F_new[all_F_new.position==p].ALT.isnull().sum())\n",
    "        else:\n",
    "            if 'LONG' in alt:\n",
    "                alt = 'X' * int(re.findall('\\d+', alt)[0])\n",
    "            F_new.set_value(pos,'count',all_F_new[(all_F_new.position == p)&(all_F_new.ALT == alt)].shape[0])\n",
    "        s = all_F_new[all_F_new.position == p].strand.dropna().unique()[0]\n",
    "        u = abs(all_F_new[all_F_new.position == p].adjustment.dropna().unique()[0])\n",
    "        o = len(r) - len(alt)\n",
    "        F_new.set_value(pos,'strand',s)\n",
    "        F_new.set_value(pos,'offset',o)\n",
    "        q = int(p - 1 + ((u % 3) + o) * s)\n",
    "        if s == 1:\n",
    "            n = MG1655[q : q + buffer]\n",
    "            aa = n.seq.translate()\n",
    "            l = len(aa.split('*')[0]) * 3\n",
    "            F_new.set_value(pos,'codon_alt',str(MG1655[q + l : q + l + 3].seq))\n",
    "        else:\n",
    "            u -= 2 # resize u for accurate reporting of upstream bases on reverse strand\n",
    "            n = MG1655[q - buffer : q]\n",
    "            aa = n.seq.reverse_complement().translate()\n",
    "            l = len(aa.split('*')[0]) * 3\n",
    "            F_new.set_value(pos,'codon_alt',str(MG1655[q - l - 3 : q - l].reverse_complement().seq))\n",
    "            l += o # resize l for accurate reporting of new downstream bases on reverse strand\n",
    "        F_new.set_value(pos,'nt_upstream',u)\n",
    "        F_new.set_value(pos,'new_downstream',l + 1)\n",
    "\n",
    "print('new frameshift mutations that impact stop codon usage:\\n',F_new)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# write out to excel\n",
    "writer = pd.ExcelWriter('StopCodons.xlsx', engine='xlsxwriter')\n",
    "matched_TAG.to_excel(writer, sheet_name = 'ancestral TAGs',startcol=0)\n",
    "unmatched_TAG.to_excel(writer, sheet_name = 'ancestral TAGs',startcol=6)\n",
    "pd.DataFrame(missing_TAG).to_excel(writer, sheet_name = 'ancestral TAGs',startcol=12)\n",
    "ancestral.to_excel(writer, sheet_name = 'ancestral TAA-TGAs')\n",
    "underhit_TAG_real.to_excel(writer, sheet_name = 'CFP TAG reversions')\n",
    "cfp_TAATGAnew.to_excel(writer, sheet_name = 'new CFP TAA-TGAs')\n",
    "e_STOPmut.to_excel(writer, sheet_name = 'new E TAA-TAG-TGAs')\n",
    "F_ancestral.to_excel(writer, sheet_name = 'ancestral FS')\n",
    "F_reversion.to_excel(writer, sheet_name = 'FS reversions')\n",
    "F_new.to_excel(writer, sheet_name = 'new FS')\n",
    "writer.save()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UID</th>\n",
       "      <th>position</th>\n",
       "      <th>lineage</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>DP</th>\n",
       "      <th>AF</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>435713</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-B5-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>32.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435714</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-F5-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>83.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435716</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-D5-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>43.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435717</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-9-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>27.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435718</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-3-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>39.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435719</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-G6-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435720</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-8-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>46.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435721</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-C5-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>27.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435722</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-A6-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>82.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435723</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-13-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>7.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435724</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-A6-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>65.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435725</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-12-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>10.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435727</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-G5-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>79.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435728</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-B6-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>47.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435729</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-G5-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>62.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435730</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-4-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435731</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-C6-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>68.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435732</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-2-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>30.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435733</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-E4-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>51.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435734</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-C5-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>69.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435735</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-5-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>19.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435736</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-D6-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>58.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435737</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-B6-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>58.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435738</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-14-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435739</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-D4-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>66.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435740</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-H5-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>56.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435741</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-5-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435743</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-14-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>19.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435744</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-1-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>7.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435745</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-13-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435746</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-G4-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435747</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-F3-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>78.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435748</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-8-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435749</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-E4-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>80.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435751</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-C3-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>30.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435752</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-G0</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>37.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435753</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-6-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435754</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-7-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>31.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435756</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-G3-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435757</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-G3-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435758</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-C4-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>51.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435759</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-C3-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>68.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435761</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-B3-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>71.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435762</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-F4-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>63.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435763</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-A4-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>52.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435764</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-G4-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>55.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435765</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-11-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>78.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435766</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-G0</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>31.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435771</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-B4-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>56.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435772</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-A4-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>40.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435773</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-C6-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>50.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435774</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-C4-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>62.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435776</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-E3-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>62.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435778</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-9-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>47.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435780</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-10-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>64.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435783</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-10-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435784</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-D3-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>43.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435788</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-D3-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>38.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435790</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-11-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435791</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-E6-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>75.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435792</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-B5-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>40.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435794</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-E5-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435795</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-D5-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>55.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435796</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-F4-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435798</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-F3-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435800</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-E3-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>69.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435801</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-E6-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>54.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435803</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-H5-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>46.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435805</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-12-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435809</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-D4-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435812</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-F5-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>33.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435813</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-F6-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>65.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435814</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-2-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>10.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435816</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-D6-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>60.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435818</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-6-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>25.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435819</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-1-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>9.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435820</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-B4-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>87.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435821</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-3-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435822</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>C-B3-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>62.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435823</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-4-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>27.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435824</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-G6-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>59.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435825</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-F6-1</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>68.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435826</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>F-E5-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>50.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435827</th>\n",
       "      <td>b5a22b1c</td>\n",
       "      <td>3862523</td>\n",
       "      <td>P-7-2</td>\n",
       "      <td>TGCCC</td>\n",
       "      <td>TC</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>glvC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             UID  position lineage    REF  ALT    DP   AF  gene\n",
       "435713  b5a22b1c   3862523  F-B5-1  TGCCC   TC  32.0  1.0  glvC\n",
       "435714  b5a22b1c   3862523  F-F5-1  TGCCC   TC  83.0  1.0  glvC\n",
       "435716  b5a22b1c   3862523  F-D5-2  TGCCC   TC  43.0  1.0  glvC\n",
       "435717  b5a22b1c   3862523   P-9-2  TGCCC   TC  27.0  1.0  glvC\n",
       "435718  b5a22b1c   3862523   P-3-1  TGCCC   TC  39.0  1.0  glvC\n",
       "435719  b5a22b1c   3862523  F-G6-2  TGCCC   TC  20.0  1.0  glvC\n",
       "435720  b5a22b1c   3862523   P-8-1  TGCCC   TC  46.0  1.0  glvC\n",
       "435721  b5a22b1c   3862523  F-C5-1  TGCCC   TC  27.0  1.0  glvC\n",
       "435722  b5a22b1c   3862523  F-A6-1  TGCCC   TC  82.0  1.0  glvC\n",
       "435723  b5a22b1c   3862523  P-13-2  TGCCC   TC   7.0  1.0  glvC\n",
       "435724  b5a22b1c   3862523  F-A6-2  TGCCC   TC  65.0  1.0  glvC\n",
       "435725  b5a22b1c   3862523  P-12-1  TGCCC   TC  10.0  1.0  glvC\n",
       "435727  b5a22b1c   3862523  F-G5-1  TGCCC   TC  79.0  1.0  glvC\n",
       "435728  b5a22b1c   3862523  F-B6-1  TGCCC   TC  47.0  1.0  glvC\n",
       "435729  b5a22b1c   3862523  F-G5-2  TGCCC   TC  62.0  1.0  glvC\n",
       "435730  b5a22b1c   3862523   P-4-1  TGCCC   TC  13.0  1.0  glvC\n",
       "435731  b5a22b1c   3862523  F-C6-2  TGCCC   TC  68.0  1.0  glvC\n",
       "435732  b5a22b1c   3862523   P-2-1  TGCCC   TC  30.0  1.0  glvC\n",
       "435733  b5a22b1c   3862523  C-E4-1  TGCCC   TC  51.0  1.0  glvC\n",
       "435734  b5a22b1c   3862523  F-C5-2  TGCCC   TC  69.0  1.0  glvC\n",
       "435735  b5a22b1c   3862523   P-5-1  TGCCC   TC  19.0  1.0  glvC\n",
       "435736  b5a22b1c   3862523  F-D6-1  TGCCC   TC  58.0  1.0  glvC\n",
       "435737  b5a22b1c   3862523  F-B6-2  TGCCC   TC  58.0  1.0  glvC\n",
       "435738  b5a22b1c   3862523  P-14-2  TGCCC   TC  14.0  1.0  glvC\n",
       "435739  b5a22b1c   3862523  C-D4-1  TGCCC   TC  66.0  1.0  glvC\n",
       "435740  b5a22b1c   3862523  F-H5-1  TGCCC   TC  56.0  1.0  glvC\n",
       "435741  b5a22b1c   3862523   P-5-2  TGCCC   TC   8.0  1.0  glvC\n",
       "435743  b5a22b1c   3862523  P-14-1  TGCCC   TC  19.0  1.0  glvC\n",
       "435744  b5a22b1c   3862523   P-1-2  TGCCC   TC   7.0  1.0  glvC\n",
       "435745  b5a22b1c   3862523  P-13-1  TGCCC   TC  18.0  1.0  glvC\n",
       "435746  b5a22b1c   3862523  C-G4-2  TGCCC   TC  70.0  1.0  glvC\n",
       "435747  b5a22b1c   3862523  C-F3-2  TGCCC   TC  78.0  1.0  glvC\n",
       "435748  b5a22b1c   3862523   P-8-2  TGCCC   TC  14.0  1.0  glvC\n",
       "435749  b5a22b1c   3862523  C-E4-2  TGCCC   TC  80.0  1.0  glvC\n",
       "435751  b5a22b1c   3862523  C-C3-2  TGCCC   TC  30.0  1.0  glvC\n",
       "435752  b5a22b1c   3862523    F-G0  TGCCC   TC  37.0  1.0  glvC\n",
       "435753  b5a22b1c   3862523   P-6-2  TGCCC   TC  15.0  1.0  glvC\n",
       "435754  b5a22b1c   3862523   P-7-1  TGCCC   TC  31.0  1.0  glvC\n",
       "435756  b5a22b1c   3862523  C-G3-2  TGCCC   TC  29.0  1.0  glvC\n",
       "435757  b5a22b1c   3862523  C-G3-1  TGCCC   TC  26.0  1.0  glvC\n",
       "435758  b5a22b1c   3862523  C-C4-1  TGCCC   TC  51.0  1.0  glvC\n",
       "435759  b5a22b1c   3862523  C-C3-1  TGCCC   TC  68.0  1.0  glvC\n",
       "435761  b5a22b1c   3862523  C-B3-1  TGCCC   TC  71.0  1.0  glvC\n",
       "435762  b5a22b1c   3862523  C-F4-2  TGCCC   TC  63.0  1.0  glvC\n",
       "435763  b5a22b1c   3862523  C-A4-2  TGCCC   TC  52.0  1.0  glvC\n",
       "435764  b5a22b1c   3862523  C-G4-1  TGCCC   TC  55.0  1.0  glvC\n",
       "435765  b5a22b1c   3862523  P-11-1  TGCCC   TC  78.0  1.0  glvC\n",
       "435766  b5a22b1c   3862523    C-G0  TGCCC   TC  31.0  1.0  glvC\n",
       "435771  b5a22b1c   3862523  C-B4-1  TGCCC   TC  56.0  1.0  glvC\n",
       "435772  b5a22b1c   3862523  C-A4-1  TGCCC   TC  40.0  1.0  glvC\n",
       "435773  b5a22b1c   3862523  F-C6-1  TGCCC   TC  50.0  1.0  glvC\n",
       "435774  b5a22b1c   3862523  C-C4-2  TGCCC   TC  62.0  1.0  glvC\n",
       "435776  b5a22b1c   3862523  C-E3-1  TGCCC   TC  62.0  1.0  glvC\n",
       "435778  b5a22b1c   3862523   P-9-1  TGCCC   TC  47.0  1.0  glvC\n",
       "435780  b5a22b1c   3862523  P-10-1  TGCCC   TC  64.0  1.0  glvC\n",
       "435783  b5a22b1c   3862523  P-10-2  TGCCC   TC  15.0  1.0  glvC\n",
       "435784  b5a22b1c   3862523  C-D3-1  TGCCC   TC  43.0  1.0  glvC\n",
       "435788  b5a22b1c   3862523  C-D3-2  TGCCC   TC  38.0  1.0  glvC\n",
       "435790  b5a22b1c   3862523  P-11-2  TGCCC   TC  26.0  1.0  glvC\n",
       "435791  b5a22b1c   3862523  F-E6-1  TGCCC   TC  75.0  1.0  glvC\n",
       "435792  b5a22b1c   3862523  F-B5-2  TGCCC   TC  40.0  1.0  glvC\n",
       "435794  b5a22b1c   3862523  F-E5-1  TGCCC   TC  36.0  1.0  glvC\n",
       "435795  b5a22b1c   3862523  F-D5-1  TGCCC   TC  55.0  1.0  glvC\n",
       "435796  b5a22b1c   3862523  C-F4-1  TGCCC   TC   8.0  1.0  glvC\n",
       "435798  b5a22b1c   3862523  C-F3-1  TGCCC   TC  20.0  1.0  glvC\n",
       "435800  b5a22b1c   3862523  C-E3-2  TGCCC   TC  69.0  1.0  glvC\n",
       "435801  b5a22b1c   3862523  F-E6-2  TGCCC   TC  54.0  1.0  glvC\n",
       "435803  b5a22b1c   3862523  F-H5-2  TGCCC   TC  46.0  1.0  glvC\n",
       "435805  b5a22b1c   3862523  P-12-2  TGCCC   TC  16.0  1.0  glvC\n",
       "435809  b5a22b1c   3862523  C-D4-2  TGCCC   TC  70.0  1.0  glvC\n",
       "435812  b5a22b1c   3862523  F-F5-2  TGCCC   TC  33.0  1.0  glvC\n",
       "435813  b5a22b1c   3862523  F-F6-2  TGCCC   TC  65.0  1.0  glvC\n",
       "435814  b5a22b1c   3862523   P-2-2  TGCCC   TC  10.0  1.0  glvC\n",
       "435816  b5a22b1c   3862523  F-D6-2  TGCCC   TC  60.0  1.0  glvC\n",
       "435818  b5a22b1c   3862523   P-6-1  TGCCC   TC  25.0  1.0  glvC\n",
       "435819  b5a22b1c   3862523   P-1-1  TGCCC   TC   9.0  1.0  glvC\n",
       "435820  b5a22b1c   3862523  C-B4-2  TGCCC   TC  87.0  1.0  glvC\n",
       "435821  b5a22b1c   3862523   P-3-2  TGCCC  NaN   NaN  0.0   NaN\n",
       "435822  b5a22b1c   3862523  C-B3-2  TGCCC   TC  62.0  1.0  glvC\n",
       "435823  b5a22b1c   3862523   P-4-2  TGCCC   TC  27.0  1.0  glvC\n",
       "435824  b5a22b1c   3862523  F-G6-1  TGCCC   TC  59.0  1.0  glvC\n",
       "435825  b5a22b1c   3862523  F-F6-1  TGCCC   TC  68.0  1.0  glvC\n",
       "435826  b5a22b1c   3862523  F-E5-2  TGCCC   TC  50.0  1.0  glvC\n",
       "435827  b5a22b1c   3862523   P-7-2  TGCCC   TC  16.0  1.0  glvC"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgdata[(mgdata.position==3862523) & (mgdata.lineage.isin(cfp_samples))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UID</th>\n",
       "      <th>position</th>\n",
       "      <th>lineage</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>DP</th>\n",
       "      <th>AF</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5322</th>\n",
       "      <td>5d671889</td>\n",
       "      <td>49787</td>\n",
       "      <td>P-14-1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>64.0</td>\n",
       "      <td>0.984375</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5342</th>\n",
       "      <td>5d671889</td>\n",
       "      <td>49787</td>\n",
       "      <td>P-14-2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>30.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5450</th>\n",
       "      <td>76f928a5</td>\n",
       "      <td>49800</td>\n",
       "      <td>C-E4-1</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>92.0</td>\n",
       "      <td>0.956522</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5452</th>\n",
       "      <td>76f928a5</td>\n",
       "      <td>49800</td>\n",
       "      <td>C-D4-1</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>70.0</td>\n",
       "      <td>0.971429</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5458</th>\n",
       "      <td>76f928a5</td>\n",
       "      <td>49800</td>\n",
       "      <td>C-E4-2</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>111.0</td>\n",
       "      <td>0.981982</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5462</th>\n",
       "      <td>76f928a5</td>\n",
       "      <td>49800</td>\n",
       "      <td>C-F4-2</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>61.0</td>\n",
       "      <td>0.934426</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5469</th>\n",
       "      <td>76f928a5</td>\n",
       "      <td>49800</td>\n",
       "      <td>P-11-2</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>22.0</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5514</th>\n",
       "      <td>76f928a5</td>\n",
       "      <td>49800</td>\n",
       "      <td>C-G4-1</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.990000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5523</th>\n",
       "      <td>76f928a5</td>\n",
       "      <td>49800</td>\n",
       "      <td>C-G4-2</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>97.0</td>\n",
       "      <td>0.979381</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5532</th>\n",
       "      <td>76f928a5</td>\n",
       "      <td>49800</td>\n",
       "      <td>C-F4-1</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5540</th>\n",
       "      <td>76f928a5</td>\n",
       "      <td>49800</td>\n",
       "      <td>C-D4-2</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>82.0</td>\n",
       "      <td>0.987805</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5653</th>\n",
       "      <td>d109f632</td>\n",
       "      <td>49827</td>\n",
       "      <td>P-11-2</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>25.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5654</th>\n",
       "      <td>d109f632</td>\n",
       "      <td>49827</td>\n",
       "      <td>P-11-1</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>123.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5741</th>\n",
       "      <td>12f50184</td>\n",
       "      <td>49829</td>\n",
       "      <td>P-6-2</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>28.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5760</th>\n",
       "      <td>12f50184</td>\n",
       "      <td>49829</td>\n",
       "      <td>P-6-1</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>37.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5785</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-G0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>60.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5798</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-E6-1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>93.0</td>\n",
       "      <td>0.989247</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5803</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-E6-2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>57.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5808</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-E5-2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>86.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5810</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-F5-2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>31.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5812</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-D5-1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>70.0</td>\n",
       "      <td>0.971429</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5814</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-D5-2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>69.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5816</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-C5-1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>44.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5821</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-G6-2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>59.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5822</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-F5-1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>99.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5824</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-D6-1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>80.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5825</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-G5-2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>68.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5826</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-C5-2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>84.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5828</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-G5-1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>96.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5832</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>P-4-1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>33.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5842</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-D6-2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>73.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5847</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-E5-1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>53.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5865</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>P-4-2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>41.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5874</th>\n",
       "      <td>b750a956</td>\n",
       "      <td>49831</td>\n",
       "      <td>F-G6-1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>96.0</td>\n",
       "      <td>0.989583</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5917</th>\n",
       "      <td>f616f19e</td>\n",
       "      <td>49847</td>\n",
       "      <td>P-1-1</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>12.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6009</th>\n",
       "      <td>f616f19e</td>\n",
       "      <td>49847</td>\n",
       "      <td>P-1-2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6029</th>\n",
       "      <td>892d7c1f</td>\n",
       "      <td>50004</td>\n",
       "      <td>P-1-1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6104</th>\n",
       "      <td>892d7c1f</td>\n",
       "      <td>50004</td>\n",
       "      <td>P-1-2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6156</th>\n",
       "      <td>1c239b3b</td>\n",
       "      <td>50044</td>\n",
       "      <td>P-2-2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>19.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6194</th>\n",
       "      <td>1c239b3b</td>\n",
       "      <td>50044</td>\n",
       "      <td>F-A6-2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>58.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6212</th>\n",
       "      <td>1c239b3b</td>\n",
       "      <td>50044</td>\n",
       "      <td>F-A6-1</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>109.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6225</th>\n",
       "      <td>1c239b3b</td>\n",
       "      <td>50044</td>\n",
       "      <td>P-2-1</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>39.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6343</th>\n",
       "      <td>a7698a29</td>\n",
       "      <td>50155</td>\n",
       "      <td>P-14-2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>31.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6348</th>\n",
       "      <td>a7698a29</td>\n",
       "      <td>50155</td>\n",
       "      <td>P-14-1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>45.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6375</th>\n",
       "      <td>11b23eba</td>\n",
       "      <td>50162</td>\n",
       "      <td>F-F6-2</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>75.0</td>\n",
       "      <td>0.973333</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6428</th>\n",
       "      <td>11b23eba</td>\n",
       "      <td>50162</td>\n",
       "      <td>F-F6-1</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>113.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6535</th>\n",
       "      <td>d6bd4475</td>\n",
       "      <td>50163</td>\n",
       "      <td>P-12-2</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6554</th>\n",
       "      <td>d6bd4475</td>\n",
       "      <td>50163</td>\n",
       "      <td>P-12-1</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6609</th>\n",
       "      <td>4b4a1169</td>\n",
       "      <td>50205</td>\n",
       "      <td>C-B3-1</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>128.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6623</th>\n",
       "      <td>4b4a1169</td>\n",
       "      <td>50205</td>\n",
       "      <td>C-D3-2</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>48.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6631</th>\n",
       "      <td>4b4a1169</td>\n",
       "      <td>50205</td>\n",
       "      <td>C-D3-1</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>44.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6704</th>\n",
       "      <td>4b4a1169</td>\n",
       "      <td>50205</td>\n",
       "      <td>C-B3-2</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>72.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6727</th>\n",
       "      <td>af5a440c</td>\n",
       "      <td>50219</td>\n",
       "      <td>C-C4-1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>66.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6736</th>\n",
       "      <td>af5a440c</td>\n",
       "      <td>50219</td>\n",
       "      <td>C-C4-2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>75.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6738</th>\n",
       "      <td>af5a440c</td>\n",
       "      <td>50219</td>\n",
       "      <td>C-B4-1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>84.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6811</th>\n",
       "      <td>af5a440c</td>\n",
       "      <td>50219</td>\n",
       "      <td>C-B4-2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>94.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6842</th>\n",
       "      <td>d5dd35d2</td>\n",
       "      <td>50221</td>\n",
       "      <td>P-8-1</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>54.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6865</th>\n",
       "      <td>d5dd35d2</td>\n",
       "      <td>50221</td>\n",
       "      <td>P-13-2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6895</th>\n",
       "      <td>d5dd35d2</td>\n",
       "      <td>50221</td>\n",
       "      <td>P-13-1</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6912</th>\n",
       "      <td>d5dd35d2</td>\n",
       "      <td>50221</td>\n",
       "      <td>P-8-2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>25.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6951</th>\n",
       "      <td>82cfc0b2</td>\n",
       "      <td>50258</td>\n",
       "      <td>C-G3-1</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>25.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6955</th>\n",
       "      <td>82cfc0b2</td>\n",
       "      <td>50258</td>\n",
       "      <td>C-G3-2</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>26.0</td>\n",
       "      <td>0.961538</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6964</th>\n",
       "      <td>82cfc0b2</td>\n",
       "      <td>50258</td>\n",
       "      <td>C-F3-2</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>83.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6974</th>\n",
       "      <td>82cfc0b2</td>\n",
       "      <td>50258</td>\n",
       "      <td>C-F3-1</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>50.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>folA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           UID  position lineage REF ALT     DP        AF  gene\n",
       "5322  5d671889     49787  P-14-1   T   C   64.0  0.984375  folA\n",
       "5342  5d671889     49787  P-14-2   T   C   30.0  1.000000  folA\n",
       "5450  76f928a5     49800  C-E4-1   A   T   92.0  0.956522  folA\n",
       "5452  76f928a5     49800  C-D4-1   A   T   70.0  0.971429  folA\n",
       "5458  76f928a5     49800  C-E4-2   A   T  111.0  0.981982  folA\n",
       "5462  76f928a5     49800  C-F4-2   A   T   61.0  0.934426  folA\n",
       "5469  76f928a5     49800  P-11-2   A   T   22.0  0.090909  folA\n",
       "5514  76f928a5     49800  C-G4-1   A   T  100.0  0.990000  folA\n",
       "5523  76f928a5     49800  C-G4-2   A   T   97.0  0.979381  folA\n",
       "5532  76f928a5     49800  C-F4-1   A   T    8.0  1.000000  folA\n",
       "5540  76f928a5     49800  C-D4-2   A   T   82.0  0.987805  folA\n",
       "5653  d109f632     49827  P-11-2   T   A   25.0  1.000000  folA\n",
       "5654  d109f632     49827  P-11-1   T   A  123.0  1.000000  folA\n",
       "5741  12f50184     49829   P-6-2   A   G   28.0  1.000000  folA\n",
       "5760  12f50184     49829   P-6-1   A   G   37.0  1.000000  folA\n",
       "5785  b750a956     49831    F-G0   T   C   60.0  1.000000  folA\n",
       "5798  b750a956     49831  F-E6-1   T   C   93.0  0.989247  folA\n",
       "5803  b750a956     49831  F-E6-2   T   C   57.0  1.000000  folA\n",
       "5808  b750a956     49831  F-E5-2   T   C   86.0  1.000000  folA\n",
       "5810  b750a956     49831  F-F5-2   T   C   31.0  1.000000  folA\n",
       "5812  b750a956     49831  F-D5-1   T   C   70.0  0.971429  folA\n",
       "5814  b750a956     49831  F-D5-2   T   C   69.0  1.000000  folA\n",
       "5816  b750a956     49831  F-C5-1   T   C   44.0  1.000000  folA\n",
       "5821  b750a956     49831  F-G6-2   T   C   59.0  1.000000  folA\n",
       "5822  b750a956     49831  F-F5-1   T   C   99.0  1.000000  folA\n",
       "5824  b750a956     49831  F-D6-1   T   C   80.0  1.000000  folA\n",
       "5825  b750a956     49831  F-G5-2   T   C   68.0  1.000000  folA\n",
       "5826  b750a956     49831  F-C5-2   T   C   84.0  1.000000  folA\n",
       "5828  b750a956     49831  F-G5-1   T   C   96.0  1.000000  folA\n",
       "5832  b750a956     49831   P-4-1   T   C   33.0  1.000000  folA\n",
       "5842  b750a956     49831  F-D6-2   T   C   73.0  1.000000  folA\n",
       "5847  b750a956     49831  F-E5-1   T   C   53.0  1.000000  folA\n",
       "5865  b750a956     49831   P-4-2   T   C   41.0  1.000000  folA\n",
       "5874  b750a956     49831  F-G6-1   T   C   96.0  0.989583  folA\n",
       "5917  f616f19e     49847   P-1-1   G   A   12.0  1.000000  folA\n",
       "6009  f616f19e     49847   P-1-2   G   A   18.0  1.000000  folA\n",
       "6029  892d7c1f     50004   P-1-1   T   C   18.0  1.000000  folA\n",
       "6104  892d7c1f     50004   P-1-2   T   C   16.0  1.000000  folA\n",
       "6156  1c239b3b     50044   P-2-2   G   A   19.0  1.000000  folA\n",
       "6194  1c239b3b     50044  F-A6-2   G   A   58.0  1.000000  folA\n",
       "6212  1c239b3b     50044  F-A6-1   G   A  109.0  1.000000  folA\n",
       "6225  1c239b3b     50044   P-2-1   G   A   39.0  1.000000  folA\n",
       "6343  a7698a29     50155  P-14-2   T   C   31.0  1.000000  folA\n",
       "6348  a7698a29     50155  P-14-1   T   C   45.0  1.000000  folA\n",
       "6375  11b23eba     50162  F-F6-2   C   T   75.0  0.973333  folA\n",
       "6428  11b23eba     50162  F-F6-1   C   T  113.0  1.000000  folA\n",
       "6535  d6bd4475     50163  P-12-2   A   G   15.0  1.000000  folA\n",
       "6554  d6bd4475     50163  P-12-1   A   G   13.0  1.000000  folA\n",
       "6609  4b4a1169     50205  C-B3-1   A   G  128.0  1.000000  folA\n",
       "6623  4b4a1169     50205  C-D3-2   A   G   48.0  1.000000  folA\n",
       "6631  4b4a1169     50205  C-D3-1   A   G   44.0  1.000000  folA\n",
       "6704  4b4a1169     50205  C-B3-2   A   G   72.0  1.000000  folA\n",
       "6727  af5a440c     50219  C-C4-1   T   C   66.0  1.000000  folA\n",
       "6736  af5a440c     50219  C-C4-2   T   C   75.0  1.000000  folA\n",
       "6738  af5a440c     50219  C-B4-1   T   C   84.0  1.000000  folA\n",
       "6811  af5a440c     50219  C-B4-2   T   C   94.0  1.000000  folA\n",
       "6842  d5dd35d2     50221   P-8-1   G   A   54.0  1.000000  folA\n",
       "6865  d5dd35d2     50221  P-13-2   G   A   14.0  1.000000  folA\n",
       "6895  d5dd35d2     50221  P-13-1   G   A   16.0  1.000000  folA\n",
       "6912  d5dd35d2     50221   P-8-2   G   A   25.0  1.000000  folA\n",
       "6951  82cfc0b2     50258  C-G3-1   C   T   25.0  1.000000  folA\n",
       "6955  82cfc0b2     50258  C-G3-2   C   T   26.0  0.961538  folA\n",
       "6964  82cfc0b2     50258  C-F3-2   C   T   83.0  1.000000  folA\n",
       "6974  82cfc0b2     50258  C-F3-1   C   T   50.0  1.000000  folA"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgdata[(mgdata.gene == 'folA')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UID</th>\n",
       "      <th>position</th>\n",
       "      <th>lineage</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>DP</th>\n",
       "      <th>AF</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>842</th>\n",
       "      <td>6c16a376</td>\n",
       "      <td>7922</td>\n",
       "      <td>F-G5-1</td>\n",
       "      <td>CAA</td>\n",
       "      <td>CAAA</td>\n",
       "      <td>72.0</td>\n",
       "      <td>0.943662</td>\n",
       "      <td>yaaJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1116</th>\n",
       "      <td>5b2c41ec</td>\n",
       "      <td>10647</td>\n",
       "      <td>F-G5-1</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>59.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>yaaW</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1316</th>\n",
       "      <td>e3ee0d63</td>\n",
       "      <td>14301</td>\n",
       "      <td>F-G5-1</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>61.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>dnaJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2885</th>\n",
       "      <td>72168f09</td>\n",
       "      <td>28269</td>\n",
       "      <td>F-G5-1</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>71.0</td>\n",
       "      <td>0.985915</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2958</th>\n",
       "      <td>ee45d76d</td>\n",
       "      <td>30616</td>\n",
       "      <td>F-G5-1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>60.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>carA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3561</th>\n",
       "      <td>8c7c87b4</td>\n",
       "      <td>35373</td>\n",
       "      <td>F-G5-1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>75.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>caiE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3894</th>\n",
       "      <td>1985685c</td>\n",
       "      <td>37903</td>\n",
       "      <td>F-G5-1</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>56.0</td>\n",
       "      <td>0.982143</td>\n",
       "      <td>caiB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4964</th>\n",
       "      <td>0ad54f64</td>\n",
       "      <td>47776</td>\n",
       "      <td>F-G5-1</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>57.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>kefC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           UID  position lineage  REF   ALT    DP        AF  gene\n",
       "842   6c16a376      7922  F-G5-1  CAA  CAAA  72.0  0.943662  yaaJ\n",
       "1116  5b2c41ec     10647  F-G5-1    A     G  59.0  1.000000  yaaW\n",
       "1316  e3ee0d63     14301  F-G5-1    C     T  61.0  1.000000  dnaJ\n",
       "2885  72168f09     28269  F-G5-1    C     T  71.0  0.985915   NaN\n",
       "2958  ee45d76d     30616  F-G5-1    T     C  60.0  1.000000  carA\n",
       "3561  8c7c87b4     35373  F-G5-1    G     C  75.0  1.000000  caiE\n",
       "3894  1985685c     37903  F-G5-1    C     T  56.0  0.982143  caiB\n",
       "4964  0ad54f64     47776  F-G5-1    G     A  57.0  1.000000  kefC"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgdata[(mgdata.lineage == 'F-G5-1')&(mgdata.position < 49764)&~(mgdata.ALT.isnull())]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Check C321 Hitchhiker mutations"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "Go through the C321 alignment to MG1655 to put together a list of hitchiker mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'lineage'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m/Users/timothywannier/anaconda/lib/python3.6/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   2392\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2393\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2394\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc (pandas/_libs/index.c:5239)\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc (pandas/_libs/index.c:5085)\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item (pandas/_libs/hashtable.c:20405)\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item (pandas/_libs/hashtable.c:20359)\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'lineage'",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-18-6dbf1b5501ff>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mhhdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"variants_C321g0toMG1655.csv\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mdtype\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m'IS_HET'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'AO'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'QA'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m}\u001b[0m \u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mhhdata\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msort_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mby\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'lineage'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mhh_avg_readdepth\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhhdata\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDP\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmean\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m print(\"total C321 gen0 to MG1655 variant calls: \",hhdata.UID.count(),\n",
      "\u001b[0;32m/Users/timothywannier/anaconda/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36msort_values\u001b[0;34m(self, by, axis, ascending, inplace, kind, na_position)\u001b[0m\n\u001b[1;32m   3282\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3283\u001b[0m             \u001b[0mby\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mby\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3284\u001b[0;31m             \u001b[0mk\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mxs\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mby\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mother_axis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3285\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mk\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mndim\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m2\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3286\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/timothywannier/anaconda/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36mxs\u001b[0;34m(self, key, axis, level, drop_level)\u001b[0m\n\u001b[1;32m   1910\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1911\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0maxis\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1912\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1913\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1914\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_consolidate_inplace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/timothywannier/anaconda/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2060\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_multilevel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2061\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2062\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_column\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2063\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2064\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_getitem_column\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/timothywannier/anaconda/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_getitem_column\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2067\u001b[0m         \u001b[0;31m# get column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2068\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_unique\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2069\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_item_cache\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2070\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2071\u001b[0m         \u001b[0;31m# duplicate columns & possible reduce dimensionality\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/timothywannier/anaconda/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_get_item_cache\u001b[0;34m(self, item)\u001b[0m\n\u001b[1;32m   1532\u001b[0m         \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcache\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1533\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mres\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1534\u001b[0;31m             \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1535\u001b[0m             \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_box_item_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1536\u001b[0m             \u001b[0mcache\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mres\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/timothywannier/anaconda/lib/python3.6/site-packages/pandas/core/internals.py\u001b[0m in \u001b[0;36mget\u001b[0;34m(self, item, fastpath)\u001b[0m\n\u001b[1;32m   3588\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3589\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0misnull\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3590\u001b[0;31m                 \u001b[0mloc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3591\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3592\u001b[0m                 \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0misnull\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/timothywannier/anaconda/lib/python3.6/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   2393\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2394\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2395\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_maybe_cast_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2396\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2397\u001b[0m         \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtolerance\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtolerance\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc (pandas/_libs/index.c:5239)\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc (pandas/_libs/index.c:5085)\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item (pandas/_libs/hashtable.c:20405)\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item (pandas/_libs/hashtable.c:20359)\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'lineage'"
     ]
    }
   ],
   "source": [
    "hhdata = pd.read_csv(\"variants_C321g0toMG1655.csv\",dtype = {'IS_HET': str, 'AO': str, 'QA': str} )\n",
    "hhdata.sort_values(by='lineage')\n",
    "hh_avg_readdepth = hhdata.DP.mean()\n",
    "\n",
    "print(\"total C321 gen0 to MG1655 variant calls: \",hhdata.UID.count(),\n",
    "      \" -- at an average read depth of: \", round(hh_avg_readdepth,1))\n",
    "print(\"variant calls under min read depth: \",hhdata[hhdata.DP < depth].UID.count())\n",
    "print(\"variant MG1655 calls under min allele fraction: \",hhdata[hhdata.AF < fraction].UID.count())\n",
    "print()\n",
    "\n",
    "hhc = hhdata[((hhdata.DP >= depth) & (hhdata.REF.str.len() < long)) |\n",
    "                 ((hhdata.DP >= depth_long) & (hhdata.REF.str.len() > long))]\n",
    "hhc = hhc[((hhc.AF >= fraction) & (hhc.REF.str.len() < long)) | \n",
    "          ((hhc.AF >= fraction_long) & (hhc.REF.str.len() > long))]\n",
    "\n",
    "print(\"total C321 gen0 to MG1655 culled variant calls: \",hhc.UID.count())\n",
    "print()\n",
    "\n",
    "hhc_noTAG = hhc[~hhc.position.isin(positions)]\n",
    "\n",
    "print(\"C321 gen0 to MG1655 culled hitchiker calls, with TAG-TAA conversions removed: \",hhc_noTAG.UID.count())\n",
    "print(\"apparent TAAs removed:\",hhc.UID.count()-hhc_noTAG.UID.count())\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Take list of hitchikers and begin analysis..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mg_HHs = mgdata[(mgdata.position.isin(hhc_noTAG.position)&(mgdata.lineage.str[0]!='E'))].loc[:,['position','lineage','gene','REF','ALT','DP','AF']]\n",
    "hh_group = mg_HHs.groupby(['position'])\n",
    "hh_altgroup = mg_HHs.groupby(['position','ALT'])\n",
    "samples = mg_HHs.lineage.nunique()\n",
    "positions = mg_HHs.position.nunique()\n",
    "variants = mg_HHs[~mg_HHs.ALT.isnull()].index.size\n",
    "                 \n",
    "underhit_hh_altgroup = hh_altgroup.filter(lambda x: len(x) < samples)\n",
    "overhit_hh_altgroup = hh_altgroup.filter(lambda x: len(x) > samples)\n",
    "underhit_hhpos = pd.DataFrame()\n",
    "overhit_hhpos = pd.DataFrame()\n",
    "\n",
    "for pos in underhit_hh_altgroup.position.unique():\n",
    "    underhit_hhpos = pd.concat([underhit_hhpos,hh_group.get_group(pos)])\n",
    "for pos in overhit_hh_altgroup.position.unique():\n",
    "    overhit_hhpos = pd.concat([overhit_hhpos,hh_group.get_group(pos)])\n",
    "\n",
    "underhit_hhs = underhit_hhpos[underhit_hhpos.ALT.isnull()&underhit_hhpos.DP.notnull()]\n",
    "#filter out one C-G0 parental mutation, 13 F11 starter mutations, and extraneous junk:\n",
    "underhit_hhs = underhit_hhs.groupby('position').filter(lambda x: len(x) < 3)\n",
    "\n",
    "print(\"There are\",positions,\"hitchiker positions found, which times\",samples,\n",
    "      \"samples aligned, should be\",samples*positions,\n",
    "      \"lines.\\nNumber of called variants is:\",variants,\", or\",\n",
    "      samples*positions-variants,\"fewer than predicted.\\n\")\n",
    "print(\"There are\",underhit_hhs.position.nunique(),\"positions hit fewer than \",samples,\n",
    "      \" times, which are: \\n\",underhit_hhs.position.unique())\n",
    "print(\"The positions UIDs hit more than \",samples,\" times are: \\n\",overhit_hh_altgroup.position.unique())\n",
    "print(\"Underhit sample positions:\\n\",underhit_hhs.gene.value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mg_TAAs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "underhit_hhs.groupby('position').filter(lambda x: len(x) < 26).position.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# Look through Dan Rice's Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "drHom = pd.read_csv(\"samples_c321_homoplasies_vx.csv\")\n",
    "compare = pd.merge(left=drHom,right=c3variants, left_on='POS', right_on='position')\n",
    "compare = compare[['POS','REF_x','REF_y','ALT_x','ALT_y','gene','Samples_with_mutation']].sort_values(by='gene').drop_duplicates()\n",
    "compare.to_csv('drc3Compare.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pseudo = False\n",
    "if not pseudo:\n",
    "    print('not')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "mgdata.loc['letter1'] = [x[0] for x in mgdata.columns.tolist()]\n",
    "mgdata.loc['letter2'] = [x[1] for x in mgdata.columns.tolist()]\n",
    "mgdata = mgdata[mgdata.columns[mgdata.loc['letter2'].argsort()]]\n",
    "mgdata = mgdata[mgdata.columns[mgdata.loc['letter1'].argsort()]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mgdata[]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
